<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="Sherlock Cluster documentation"><meta name=author content="Kilian Cavalotti"><link href=https://www.sherlock.stanford.edu/docs/software/list/ rel=canonical><link rel="shortcut icon" href=/assets/img/logo_small.png><meta name=generator content="mkdocs-1.1.2, mkdocs-material-6.2.8"><title>List - Sherlock</title><link rel=stylesheet href=../../../assets/stylesheets/main.cb6bc1d0.min.css><link rel=stylesheet href=../../../assets/stylesheets/palette.39b8e14a.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback"><style>body,input{font-family:"Roboto",-apple-system,BlinkMacSystemFont,Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono",SFMono-Regular,Consolas,Menlo,monospace}</style><link rel=manifest href=../../../manifest.webmanifest crossorigin=use-credentials><link rel=stylesheet href=/assets/css/sh.css><script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-68034405-2","auto"),ga("set","anonymizeIp",!0),ga("send","pageview"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){if(this.value){var e=document.location.pathname;ga("send","pageview",e+"?q="+this.value)}})}),document.addEventListener("DOMContentSwitch",function(){ga("send","pageview",document.location.pathname)})</script><script async src=https://www.google-analytics.com/analytics.js></script><!-- Track external links --><script>
    document.addEventListener("DOMContentLoaded", function() {
      document.body.addEventListener("click", function(ev) {
        if (ev.target instanceof HTMLElement) {
          var el = ev.target.closest("a[href^=http]")
          if (el)
            ga("send", "event", "outbound", "click", el.href)
        }
      })
    })
  </script><!-- Determine title --><!-- The image needs to have an absolute URL --><!-- Open graph meta tags --><meta property=og:type content=website><meta property=og:title content="Sherlock - List"><meta property=og:description content="Sherlock Cluster documentation"><meta content=https://www.sherlock.stanford.edu/docs/software/list/ property=og:url><meta property=og:image content=https://www.sherlock.stanford.edu/assets/img/banner.png><meta property=og:image:type content=image/png><meta property=og:image:width content=1200><meta property=og:image:height content=630><!-- Twitter meta tags --><meta name=twitter:card content=summary_large_image><meta name=twitter:site content=@StanfordCompute><meta name=twitter:creator content=@StanfordCompute><meta name=twitter:title content="Sherlock - List"><meta name=twitter:description content="Sherlock Cluster documentation"><meta name=twitter:image content=https://www.sherlock.stanford.edu/assets/img/banner.png></head> <body dir=ltr data-md-color-scheme data-md-color-primary=none data-md-color-accent=none> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#software-list class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class=md-header data-md-component=header> <nav class="md-header-nav md-grid" aria-label=Header> <a href=https://www.sherlock.stanford.edu/ title=Sherlock class="md-header-nav__button md-logo" aria-label=Sherlock> <img src=/assets/img/logo_small.png alt=logo> </a> <label class="md-header-nav__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg> </label> <div class=md-header-nav__title data-md-component=header-title> <div class=md-header-nav__ellipsis> <div class=md-header-nav__topic> <span class=md-ellipsis> Sherlock </span> </div> <div class=md-header-nav__topic> <span class=md-ellipsis> List </span> </div> </div> </div> <label class="md-header-nav__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query data-md-state=active required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg> </label> <button type=reset class="md-search__icon md-icon" aria-label=Clear data-md-component=search-reset tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg> </button> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list></ol> </div> </div> </div> </div> </div> <div class=md-header-nav__source> <a href=https://github.com/stanford-rc/ title="Go to repository" class=md-source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg> </div> <div class=md-source__repository> SRCC // Open Source </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=https://www.sherlock.stanford.edu/ title=Sherlock class="md-nav__button md-logo" aria-label=Sherlock> <img src=/assets/img/logo_small.png alt=logo> </a> Sherlock </label> <div class=md-nav__source> <a href=https://github.com/stanford-rc/ title="Go to repository" class=md-source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg> </div> <div class=md-source__repository> SRCC // Open Source </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=nav-1 type=checkbox id=nav-1> <label class=md-nav__link for=nav-1> Overview <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Overview data-md-level=1> <label class=md-nav__title for=nav-1> <span class="md-nav__icon md-icon"></span> Overview </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../overview/introduction/ class=md-nav__link> Introduction </a> </li> <li class=md-nav__item> <a href=../../overview/concepts/ class=md-nav__link> Concepts </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=nav-1-3 type=checkbox id=nav-1-3> <label class=md-nav__link for=nav-1-3> Ordering nodes <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label="Ordering nodes" data-md-level=2> <label class=md-nav__title for=nav-1-3> <span class="md-nav__icon md-icon"></span> Ordering nodes </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../overview/orders/ class=md-nav__link> Process </a> </li> <li class=md-nav__item> <a href=/catalog/ class=md-nav__link> Catalog </a> </li> <li class=md-nav__item> <a href=/order_form/ class=md-nav__link> Order form </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=nav-1-4 type=checkbox id=nav-1-4> <label class=md-nav__link for=nav-1-4> Technical <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Technical data-md-level=2> <label class=md-nav__title for=nav-1-4> <span class="md-nav__icon md-icon"></span> Technical </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../overview/tech/facts/ class=md-nav__link> Facts </a> </li> <li class=md-nav__item> <a href=../../overview/tech/specs/ class=md-nav__link> Specs </a> </li> <li class=md-nav__item> <a href=../../overview/tech/status/ class=md-nav__link> Status </a> </li> <li class=md-nav__item> <a href=../../overview/tech/glossary/ class=md-nav__link> Glossary </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../../overview/about/ class=md-nav__link> About </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=nav-2 type=checkbox id=nav-2> <label class=md-nav__link for=nav-2> Getting Started <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label="Getting Started" data-md-level=1> <label class=md-nav__title for=nav-2> <span class="md-nav__icon md-icon"></span> Getting Started </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../getting-started/prerequisites/ class=md-nav__link> Prerequisites </a> </li> <li class=md-nav__item> <a href=../../getting-started/connecting/ class=md-nav__link> Connecting </a> </li> <li class=md-nav__item> <a href=../../getting-started/submitting/ class=md-nav__link> Submitting jobs </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=nav-3 type=checkbox id=nav-3> <label class=md-nav__link for=nav-3> User guide <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label="User guide" data-md-level=1> <label class=md-nav__title for=nav-3> <span class="md-nav__icon md-icon"></span> User guide </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../user-guide/running-jobs/ class=md-nav__link> Running jobs </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=nav-3-2 type=checkbox id=nav-3-2> <label class=md-nav__link for=nav-3-2> Specialized resources <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label="Specialized resources" data-md-level=2> <label class=md-nav__title for=nav-3-2> <span class="md-nav__icon md-icon"></span> Specialized resources </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../user-guide/gpu/ class=md-nav__link> GPU nodes </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../../user-guide/ondemand/ class=md-nav__link> OnDemand </a> </li> <li class=md-nav__item> <a href=../../user-guide/troubleshoot/ class=md-nav__link> Troubleshooting </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=nav-4 type=checkbox id=nav-4> <label class=md-nav__link for=nav-4> Storage <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Storage data-md-level=1> <label class=md-nav__title for=nav-4> <span class="md-nav__icon md-icon"></span> Storage </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../storage/overview/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../../storage/filesystems/ class=md-nav__link> Filesystems </a> </li> <li class=md-nav__item> <a href=../../storage/data-sharing/ class=md-nav__link> Data sharing </a> </li> <li class=md-nav__item> <a href=../../storage/data-protection/ class=md-nav__link> Data protection </a> </li> <li class=md-nav__item> <a href=../../storage/data-transfer/ class=md-nav__link> Data transfer </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--active md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=nav-5 type=checkbox id=nav-5 checked> <label class=md-nav__link for=nav-5> Software <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Software data-md-level=1> <label class=md-nav__title for=nav-5> <span class="md-nav__icon md-icon"></span> Software </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../overview/ class=md-nav__link> Overview </a> </li> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" data-md-toggle=toc type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> List <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> List </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=#categories class=md-nav__link> Categories </a> <nav class=md-nav aria-label=Categories> <ul class=md-nav__list> <li class=md-nav__item> <a href=#biology class=md-nav__link> biology </a> </li> <li class=md-nav__item> <a href=#chemistry class=md-nav__link> chemistry </a> </li> <li class=md-nav__item> <a href=#devel class=md-nav__link> devel </a> </li> <li class=md-nav__item> <a href=#math class=md-nav__link> math </a> </li> <li class=md-nav__item> <a href=#physics class=md-nav__link> physics </a> </li> <li class=md-nav__item> <a href=#system class=md-nav__link> system </a> </li> <li class=md-nav__item> <a href=#viz class=md-nav__link> viz </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../modules/ class=md-nav__link> Modules </a> </li> <li class=md-nav__item> <a href=../install/ class=md-nav__link> Installation </a> </li> <li class=md-nav__item> <a href=../using/singularity/ class=md-nav__link> Containers </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=nav-5-6 type=checkbox id=nav-5-6> <label class=md-nav__link for=nav-5-6> Usage <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Usage data-md-level=2> <label class=md-nav__title for=nav-5-6> <span class="md-nav__icon md-icon"></span> Usage </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../using/R/ class=md-nav__link> R </a> </li> <li class=md-nav__item> <a href=../using/julia/ class=md-nav__link> Julia </a> </li> <li class=md-nav__item> <a href=../using/python/ class=md-nav__link> Python </a> </li> <li class=md-nav__item> <a href=../using/perl/ class=md-nav__link> Perl </a> </li> <li class=md-nav__item> <a href=../using/matlab/ class=md-nav__link> Matlab </a> </li> <li class=md-nav__item> <a href=../using/rclone/ class=md-nav__link> Rclone </a> </li> <li class=md-nav__item> <a href=../using/singularity/ class=md-nav__link> Singularity </a> </li> <li class=md-nav__item> <a href=../using/spark/ class=md-nav__link> Spark </a> </li> <li class=md-nav__item> <a href=../using/mariadb/ class=md-nav__link> MariaDB </a> </li> <li class=md-nav__item> <a href=../using/postgresql/ class=md-nav__link> PostgreSQL </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=nav-6 type=checkbox id=nav-6> <label class=md-nav__link for=nav-6> Advanced topics <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label="Advanced topics" data-md-level=1> <label class=md-nav__title for=nav-6> <span class="md-nav__icon md-icon"></span> Advanced topics </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../advanced-topics/connection/ class=md-nav__link> Connection </a> </li> <li class=md-nav__item> <a href=../../advanced-topics/job-management/ class=md-nav__link> Job management </a> </li> </ul> </nav> </li> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=#categories class=md-nav__link> Categories </a> <nav class=md-nav aria-label=Categories> <ul class=md-nav__list> <li class=md-nav__item> <a href=#biology class=md-nav__link> biology </a> </li> <li class=md-nav__item> <a href=#chemistry class=md-nav__link> chemistry </a> </li> <li class=md-nav__item> <a href=#devel class=md-nav__link> devel </a> </li> <li class=md-nav__item> <a href=#math class=md-nav__link> math </a> </li> <li class=md-nav__item> <a href=#physics class=md-nav__link> physics </a> </li> <li class=md-nav__item> <a href=#system class=md-nav__link> system </a> </li> <li class=md-nav__item> <a href=#viz class=md-nav__link> viz </a> </li> </ul> </nav> </li> </ul> </nav> </div> </div> </div> <div class=md-content> <article class="md-content__inner md-typeset"> <a href=https://github.com/stanford-rc/www.sherlock.stanford.edu/edit/main/src/docs/software/list.md title="Edit this page" class="md-content__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg> </a> <h1 id=software-list>Software list<a class=headerlink href=#software-list title="Permanent link">#</a></h1> <p>The full list of software centrally installed and managed on Sherlock is in the tables below.</p> <div class="admonition todo"> <p class=admonition-title>Work in progress</p> <p>Software installations on Sherlock are an ever ongoing process. We're continuously adding new software to the list. If you're looking for something that is not in the list, please take a look <a href=/docs/software#installation-requests>here</a> for options.</p> </div> <h2 id=categories>Categories<a class=headerlink href=#categories title="Permanent link">#</a></h2> <p>Software modules on Sherlock are organized in <em>categories</em>, by scientific field or functional class. It means that you will have to first load a category module before getting access to individual modules. The <code>math</code> and <code>devel</code> categories are loaded by default. See the <a href=/docs/software/modules/ >Modules page</a> for further details and examples.</p> <p><em>We currently provide 416 software modules, in 7 categories, covering 75 fields of science:</em></p> <!-- markdownlint-disable MD032 --> <ul> <li> <p><a href=#biology><code>biology</code></a> <small> clinical science, computational biology, cryo-em, genomics, neurology, pathology, phylogenetics, population genetics </small></p> </li> <li> <p><a href=#chemistry><code>chemistry</code></a> <small> cheminformatics, computational chemistry, crystallography, electrostatics, molecular dynamics, quantum chemistry </small></p> </li> <li> <p><a href=#devel><code>devel</code></a> <small> build, compiler, data, data analytics, debug, engine, framework, language, lib, mpi, networking, parser, profiling, runtime, sdk </small></p> </li> <li> <p><a href=#math><code>math</code></a> <small> computational geometry, deep learning, linear algebra, machine learning, numerical analysis, numerical library, optimization, scientific computing, statistics, symbolic </small></p> </li> <li> <p><a href=#physics><code>physics</code></a> <small> astronomy, CFD, climate modeling, geophysics, geoscience, materials science, micromagnetics, particle, photonics, quantum mechanics </small></p> </li> <li> <p><a href=#system><code>system</code></a> <small> backup, benchmark, checkpointing, compression, containers, database, devel, scm, document management, document processing, file management, file transfer, framework, language, libs, media, performance, resource monitoring, scm, tools </small></p> </li> <li> <p><a href=#viz><code>viz</code></a> <small> data, gis, graphs, imaging, molecular visualization, plotting, remote display </small></p> </li> </ul> <!-- markdownlint-enable MD032 --> <div class="admonition warning"> <p class=admonition-title>Licensed software</p> <p>Access to software modules marked with <sup><b class=sw_lic></b></sup> in the tables below is restricted to properly licensed user groups.</p> <p><abbr title="Stanford Research Computing Center">SRCC</abbr> is not funded to provide commercial software on Sherlock and researchers are responsible for the costs of purchasing and renewing commercial software licenses. For more information, please feel free to <a href=mailto:srcc-support@stanford.edu>contact us</a> and see the <a href=https://uit.stanford.edu/service/softwarelic>Stanford Software Licensing page</a> for purchasing information.</p> </div> <div class="admonition tip"> <p class=admonition-title>Additional flags and features</p> <p>Some of the modules listed below have been built to support specific architectures or parallel execution modes:</p> <ul> <li>versions marked with <sup><b class=sw_gpu></b></sup> support <abbr title="Graphical Processing Unit">GPU</abbr> acceleration</li> <li>versions marked with <sup><b class=sw_mpi></b></sup> support <abbr title="Message Passing Interface">MPI</abbr> parallel execution</li> <li>versions marked with <sup><b class=sw_def></b></sup> are the default version for the module</li> </ul> </div> <!-- markdownlint-disable MD013 --> <!-- markdownlint-enable MD013 --> <!-- markdownlint-disable MD038 --> <!-- markdownlint-enable MD038 --> <h3 id=biology><strong>biology</strong><a class=headerlink href=#biology title="Permanent link">#</a></h3> <table> <thead> <tr> <th align=left>Field</th> <th align=left><img style=float:left;min-width:110px;visibility:hidden>Module&nbsp;name</th> <th align=left><img style=float:left;min-width:90px;visibility:hidden>Version(s)</th> <th align=left>URL</th> <th align=left>Description</th> </tr> </thead> <tbody> <tr> <td align=left><strong>clinical science</strong></td> <td align=left><a id=simvascular><code>simvascular</code></a></td> <td align=left><code>20180704</code><br></td> <td align=left><a href=https://simvascular.github.io>Website</a></td> <td align=left>Simvascular is a blood flow simulation and analysis toolkit. This module provides the svFSI (Fluid Solid Interaction) solver.</td> </tr> <tr> <td align=left><strong>computational biology</strong></td> <td align=left><a id=py-biopython><code>py-biopython</code></a></td> <td align=left><code>1.70_py27</code><br></td> <td align=left><a href=http://biopython.org>Website</a></td> <td align=left>Biopython is a set of freely available tools for biological computation written in Python.</td> </tr> <tr> <td align=left><strong>computational biology</strong></td> <td align=left><a id=rosetta><code>rosetta</code></a></td> <td align=left><code>3.8</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://www.rosettacommons.org>Website</a></td> <td align=left>Rosetta is the premier software suite for modeling macromolecular structures. As a flexible, multi-purpose application, it includes tools for structure prediction, design, and remodeling of proteins and nucleic acids.</td> </tr> <tr> <td align=left><strong>cryo-em</strong></td> <td align=left><a id=ctffind><code>ctffind</code></a></td> <td align=left><code>4.1.13</code><br></td> <td align=left><a href=https://grigoriefflab.umassmed.edu/ctf_estimation_ctffind_ctftilt>Website</a></td> <td align=left>ctffind is a program for finding CTFs of electron micrographs.</td> </tr> <tr> <td align=left><strong>cryo-em</strong></td> <td align=left><a id=eman2><code>eman2</code></a></td> <td align=left><code>2.2</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=http://blake.bcm.edu/emanwiki/EMAN2>Website</a></td> <td align=left>EMAN2 is a broadly based greyscale scientific image processing suite with a primary focus on processing data from transmission electron microscopes.</td> </tr> <tr> <td align=left><strong>cryo-em</strong></td> <td align=left><a id=imod><code>imod</code></a></td> <td align=left><code>4.9.12</code><sup><b class=sw_gpu></b></sup><br><code>4.11.5</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=http://bio3d.colorado.edu/imod>Website</a></td> <td align=left>IMOD is a set of image processing, modeling and display programs used for tomographic reconstruction and for 3D reconstruction of EM serial sections and optical sections.</td> </tr> <tr> <td align=left><strong>cryo-em</strong></td> <td align=left><a id=motioncor2><code>motioncor2</code></a></td> <td align=left><code>1.3.1</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://emcore.ucsf.edu/ucsf-motioncor2>Website</a></td> <td align=left>MotionCor2 is a multi-<abbr title="Graphical Processing Unit">GPU</abbr> accelerated program which corrects anisotropic image motion at the single pixel level.</td> </tr> <tr> <td align=left><strong>cryo-em</strong></td> <td align=left><a id=relion><code>relion</code></a></td> <td align=left><code>2.0.3</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br><code>2.1</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=http://www2.mrc-lmb.cam.ac.uk/relion/index.php/Main_Page>Website</a></td> <td align=left>RELION (for REgularised LIkelihood OptimisatioN, pronounce rely-on) is a stand-alone computer program that employs an empirical Bayesian approach to refinement of (multiple) 3D reconstructions or 2D class averages in electron cryo-microscopy (cryo-EM).</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=angsd><code>angsd</code></a></td> <td align=left><code>0.919</code><br><code>0.931</code><br></td> <td align=left><a href=http://www.popgen.dk/angsd/index.php/ANGSD>Website</a></td> <td align=left>ANGSD is a software for analyzing next generation sequencing data.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=augustus><code>augustus</code></a></td> <td align=left><code>3.3.2</code><br></td> <td align=left><a href=https://github.com/Gaius-Augustus/Augustus>Website</a></td> <td align=left>AUGUSTUS is a program that predicts genes in eukaryotic genomic sequences.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=bamtools><code>bamtools</code></a></td> <td align=left><code>2.5.1</code><br></td> <td align=left><a href=https://github.com/pezmaster31/bamtools>Website</a></td> <td align=left>BamTools is a project that provides both a C++ API and a command-line toolkit for reading, writing, and manipulating BAM (genome alignment) files.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=bcftools><code>bcftools</code></a></td> <td align=left><code>1.6</code><br><code>1.8</code><br></td> <td align=left><a href=https://github.com/samtools/bcftools>Website</a></td> <td align=left>BCFtools is a program for variant calling and manipulating files in the Variant Call Format (VCF) and its binary counterpart BCF.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=bcl2fastq><code>bcl2fastq</code></a></td> <td align=left><code>2.20</code><br></td> <td align=left><a href=https://support.illumina.com/downloads/bcl2fastq-conversion-software-v2-20.html>Website</a></td> <td align=left>The bcl2fastq2 conversion software can be used to convert BCL files from MiniSeq, MiSeq, NextSeq, HiSeq, iSeq and NovaSeq sequening systems.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=bedtools><code>bedtools</code></a></td> <td align=left><code>2.27.1</code><br></td> <td align=left><a href=https://bedtools.readthedocs.io>Website</a></td> <td align=left>The bedtools utilities are a swiss-army knife of tools for a wide-range of genomics analysis tasks.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=bgen><code>bgen</code></a></td> <td align=left><code>1.1.4</code><br></td> <td align=left><a href=https://bitbucket.org/gavinband/bgen/ >Website</a></td> <td align=left>bgen is the reference implementation of the BGEN format, a binary file format for imputed genotype and haplotype data.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=bowtie><code>bowtie</code></a></td> <td align=left><code>1.2.2</code><br></td> <td align=left><a href=http://bowtie-bio.sourceforge.net/ >Website</a></td> <td align=left>Bowtie is an ultrafast, memory-efficient short read aligner.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=bowtie2><code>bowtie2</code></a></td> <td align=left><code>2.3.4.1</code><br></td> <td align=left><a href=http://bowtie-bio.sourceforge.net/bowtie2>Website</a></td> <td align=left>Bowtie 2 is an ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=bwa><code>bwa</code></a></td> <td align=left><code>0.7.17</code><br></td> <td align=left><a href=http://bio-bwa.sourceforge.net/ >Website</a></td> <td align=left>BWA (Burrows-Wheeler Aligner) is a software package for mapping low-divergent sequences against a large reference genome, such as the human genome.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=canu><code>canu</code></a></td> <td align=left><code>1.8</code><br></td> <td align=left><a href=https://github.com/marbl/canu/ >Website</a></td> <td align=left>A single molecule sequence assembler for genomes large and small.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=cufflinks><code>cufflinks</code></a></td> <td align=left><code>2.2.1</code><br></td> <td align=left><a href=https://cole-trapnell-lab.github.io/cufflinks/ >Website</a></td> <td align=left>Cufflinks assembles transcripts, estimates their abundances, and tests for differential expression and regulation in RNA-Seq samples.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=fastqc><code>fastqc</code></a></td> <td align=left><code>0.11.8</code><br></td> <td align=left><a href=https://www.bioinformatics.babraham.ac.uk/projects/fastqc/ >Website</a></td> <td align=left>FastQC aims to provide a simple way to do some quality control checks on raw sequence data coming from high throughput sequencing pipelines.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=fastx_toolkit><code>fastx_toolkit</code></a></td> <td align=left><code>0.0.14</code><br></td> <td align=left><a href=http://hannonlab.cshl.edu/fastx_toolkit/ >Website</a></td> <td align=left>The FASTX-Toolkit is a collection of command line tools for Short-Reads FASTA/FASTQ files preprocessing.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=freebayes><code>freebayes</code></a></td> <td align=left><code>1.2.0</code><br></td> <td align=left><a href=https://github.com/ekg/freebayes>Website</a></td> <td align=left>FreeBayes is a Bayesian genetic variant detector designed to find small polymorphisms.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=gatk><code>gatk</code></a></td> <td align=left><code>4.1.0.0</code><br><code>4.1.4.1</code><br></td> <td align=left><a href=https://software.broadinstitute.org/gatk/ >Website</a></td> <td align=left>GATK (Genome Analysis Toolkit) offers a wide variety of tools with a primary focus on variant discovery and genotyping.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=hic-pro><code>hic-pro</code></a></td> <td align=left><code>2.10.0</code><br></td> <td align=left><a href=https://github.com/nservant/HiC-Pro>Website</a></td> <td align=left>HiC-Pro: An optimized and flexible pipeline for Hi-C data processing.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=hisat2><code>hisat2</code></a></td> <td align=left><code>2.1.0</code><br></td> <td align=left><a href=http://ccb.jhu.edu/software/hisat2>Website</a></td> <td align=left>HISAT2 is a fast and sensitive alignment program for mapping next-generation sequencing reads (both DNA and RNA) to a population of human genomes (as well as to a single reference genome).</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=htslib><code>htslib</code></a></td> <td align=left><code>1.6</code><br><code>1.8</code><br><code>1.10.2</code><br></td> <td align=left><a href=https://github.com/samtools/htslib>Website</a></td> <td align=left>C library for high-throughput sequencing data formats.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=jellyfish><code>jellyfish</code></a></td> <td align=left><code>2.2.10</code><br></td> <td align=left><a href=https://github.com/gmarcais/Jellyfish>Website</a></td> <td align=left>A fast multi-threaded k-mer counter.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=kallisto><code>kallisto</code></a></td> <td align=left><code>0.44.0</code><br></td> <td align=left><a href=https://pachterlab.github.io/kallisto/ >Website</a></td> <td align=left>kallisto is a program for quantifying abundances of transcripts from RNA-Seq data using high-throughput sequencing reads.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=metal><code>metal</code></a></td> <td align=left><code>20110325</code><br></td> <td align=left><a href=http://www.sph.umich.edu/csg/abecasis/Metal/ >Website</a></td> <td align=left>The METAL software is designed to facilitate meta-analysis of large datasets (such as several whole genome scans) in a convenient, rapid and memory efficient manner.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=mixcr><code>mixcr</code></a></td> <td align=left><code>2.1.12</code><br></td> <td align=left><a href=https://milaboratory.com/software/mixcr/ >Website</a></td> <td align=left>MiXCR is a universal framework that processes big immunome data from raw sequences to quantitated clonotypes.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=ncbi-blast+><code>ncbi-blast+</code></a></td> <td align=left><code>2.6.0</code><br><code>2.7.1</code><br><code>2.11.0</code><br></td> <td align=left><a href=https://blast.ncbi.nlm.nih.gov/Blast.cgi>Website</a></td> <td align=left>NCBI BLAST+ is a suite of command-line tools to run BLAST (Basic Local Alignment Search Tool), an algorithm for comparing primary biological sequence information.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=plink><code>plink</code></a></td> <td align=left><code>1.07</code><br><code>1.90b5.3</code><br><code>2.0a1</code><br><code>2.0a2</code><br></td> <td align=left><a href=http://zzz.bwh.harvard.edu/plink/ >Website</a></td> <td align=left>PLINK is a free, open-source whole genome association analysis toolset, designed to perform a range of basic, large-scale analyses in a computationally efficient manner.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=py-busco><code>py-busco</code></a></td> <td align=left><code>3.0.2_py27</code><br></td> <td align=left><a href=https://busco.ezlab.org/ >Website</a></td> <td align=left>Assessing genome assembly and annotation completeness with Benchmarking Universal Single-Copy Orthologs (BUSCO).</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=py-bx-python><code>py-bx-python</code></a></td> <td align=left><code>0.8.1_py27</code><br></td> <td align=left><a href=https://github.com/bxlab/bx-python>Website</a></td> <td align=left>Tools for manipulating biological data, particularly multiple sequence alignments.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=py-cutadapt><code>py-cutadapt</code></a></td> <td align=left><code>1.18_py27</code><sup><b class=sw_def></b></sup><br><code>1.18_py36</code><br></td> <td align=left><a href=https://cutadapt.readthedocs.io>Website</a></td> <td align=left>Cutadapt finds and removes adapter sequences, primers, poly-A tails and other types of unwanted sequence from your high-throughput sequencing reads.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=py-deeptools><code>py-deeptools</code></a></td> <td align=left><code>3.3.1_py36</code><br></td> <td align=left><a href=https://github.com/deeptools/deepTools/ >Website</a></td> <td align=left>Tools to process and analyze deep sequencing data.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=py-fithic><code>py-fithic</code></a></td> <td align=left><code>1.1.3_py27</code><br></td> <td align=left><a href=https://github.com/ay-lab/fithic>Website</a></td> <td align=left>Fit-Hi-C is a tool for assigning statistical confidence estimates to chromosomal contact maps produced by genome architecture assays.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=py-macs2><code>py-macs2</code></a></td> <td align=left><code>2.1.1_py27</code><br></td> <td align=left><a href=https://github.com/taoliu/MACS>Website</a></td> <td align=left>MACS (Model-based Analysis of ChIP-Seq) implements a novel ChIP-Seq analysis method.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=py-mapdamage><code>py-mapdamage</code></a></td> <td align=left><code>2.2.1_py36</code><br></td> <td align=left><a href=https://github.com/ginolhac/mapDamage>Website</a></td> <td align=left>mapDamage2 is a computational framework which tracks and quantifies DNA damage patterns among ancient DNA sequencing reads generated by Next-Generation Sequencing platforms.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=py-multiqc><code>py-multiqc</code></a></td> <td align=left><code>1.6_py27</code><sup><b class=sw_def></b></sup><br><code>1.6_py36</code><br></td> <td align=left><a href=https://multiqc.info>Website</a></td> <td align=left>MultiQC is a reporting tool that parses summary statistics from results and log files generated by other bioinformatics tools.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=py-obitools><code>py-obitools</code></a></td> <td align=left><code>1.2.13_py27</code><br></td> <td align=left><a href=https://pythonhosted.org/OBITools/index.html>Website</a></td> <td align=left>OBITools is a set of programs designed for analyzing NGS data in a DNA metabarcoding context.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=py-pybedtools><code>py-pybedtools</code></a></td> <td align=left><code>0.8.0_py27</code><br><code>0.8.2_py36</code><br></td> <td align=left><a href=https://github.com/daler/pybedtools>Website</a></td> <td align=left>Pybedtools wraps and extends BEDTools and offers feature-level manipulations from within Python.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=py-pysam><code>py-pysam</code></a></td> <td align=left><code>0.14.1_py27</code><br><code>0.15.3_py36</code><br></td> <td align=left><a href=https://github.com/pysam-developers/pysam>Website</a></td> <td align=left>Pysam is a python module for reading, manipulating and writing genomic data sets.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=rsem><code>rsem</code></a></td> <td align=left><code>1.3.3</code><br></td> <td align=left><a href=https://deweylab.github.io/RSEM/ >Website</a></td> <td align=left>RSEM is a software package for estimating gene and isoform expression levels from RNA-Seq data.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=salmon><code>salmon</code></a></td> <td align=left><code>0.12.0</code><br></td> <td align=left><a href=https://combine-lab.github.io/salmon>Website</a></td> <td align=left>Highly-accurate &amp; wicked fast transcript-level quantification from RNA-seq reads using lightweight alignments.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=samtools><code>samtools</code></a></td> <td align=left><code>1.6</code><br><code>1.8</code><br></td> <td align=left><a href=https://github.com/samtools/samtools>Website</a></td> <td align=left>Tools (written in C using htslib) for manipulating next-generation sequencing data.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=sentieon><code>sentieon</code></a></td> <td align=left><code>201808.01</code><sup><b class=sw_lic></b></sup><br></td> <td align=left><a href=https://www.sentieon.com/ >Website</a></td> <td align=left>Sentieon Genomics software is a set of software tools that perform analysis of genomic data obtained from DNA sequencing.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=shapeit><code>shapeit</code></a></td> <td align=left><code>4.0.0</code><br></td> <td align=left><a href=https://odelaneau.github.io/shapeit4/ >Website</a></td> <td align=left>SHAPEIT4 is a fast and accurate method for estimation of haplotypes (aka phasing) for SNP array and high coverage sequencing data.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=star><code>star</code></a></td> <td align=left><code>2.5.4b</code><br></td> <td align=left><a href=https://github.com/alexdobin/STAR>Website</a></td> <td align=left>STAR: ultrafast universal RNA-seq aligner.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=tophat><code>tophat</code></a></td> <td align=left><code>2.1.1</code><br></td> <td align=left><a href=http://ccb.jhu.edu/software/tophat/index.shtml>Website</a></td> <td align=left>TopHat is a fast splice junction mapper for RNA-Seq reads.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=trim_galore><code>trim_galore</code></a></td> <td align=left><code>0.5.0</code><br></td> <td align=left><a href=https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/ >Website</a></td> <td align=left>Trim Galore! is a wrapper script to automate quality and adapter trimming as well as quality control, with some added functionality to remove biased methylation positions for RRBS sequence files.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=trinity><code>trinity</code></a></td> <td align=left><code>2.8.4</code><br></td> <td align=left><a href=http://trinityrnaseq.github.io>Website</a></td> <td align=left>Trinity RNA-Seq de novo transcriptome assembly.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=vcflib><code>vcflib</code></a></td> <td align=left><code>1.0.0</code><br></td> <td align=left><a href=https://github.com/vcflib/vcflib>Website</a></td> <td align=left>A C++ library for parsing and manipulating VCF files.</td> </tr> <tr> <td align=left><strong>genomics</strong></td> <td align=left><a id=vcftools><code>vcftools</code></a></td> <td align=left><code>0.1.15</code><br></td> <td align=left><a href=https://vcftools.github.io>Website</a></td> <td align=left>VCFtools is a program package designed for working with VCF files, such as those generated by the 1000 Genomes Project.</td> </tr> <tr> <td align=left><strong>neurology</strong></td> <td align=left><a id=afni><code>afni</code></a></td> <td align=left><code>17.2.07</code><br><code>18.2.04</code><br></td> <td align=left><a href=https://afni.nimh.nih.gov>Website</a></td> <td align=left>AFNI (Analysis of Functional NeuroImages) is a set of C programs for processing, analyzing, and displaying functional MRI (FMRI) data - a technique for mapping human brain activity.</td> </tr> <tr> <td align=left><strong>neurology</strong></td> <td align=left><a id=ants><code>ants</code></a></td> <td align=left><code>2.1.0</code><br><code>2.3.1</code><br></td> <td align=left><a href=https://stnava.github.io/ANTs>Website</a></td> <td align=left>ANTs computes high-dimensional mappings to capture the statistics of brain structure and function.</td> </tr> <tr> <td align=left><strong>neurology</strong></td> <td align=left><a id=dcm2niix><code>dcm2niix</code></a></td> <td align=left><code>1.0.20171215</code><br></td> <td align=left><a href=https://github.com/rordenlab/dcm2niix>Website</a></td> <td align=left>dcm2niix is a program esigned to convert neuroimaging data from the DICOM format to the NIfTI format.</td> </tr> <tr> <td align=left><strong>neurology</strong></td> <td align=left><a id=freesurfer><code>freesurfer</code></a></td> <td align=left><code>6.0.0</code><sup><b class=sw_def></b></sup><br><code>7.1.1</code><br></td> <td align=left><a href=https://surfer.nmr.mgh.harvard.edu/ >Website</a></td> <td align=left>An open source software suite for processing and analyzing (human) brain MRI images.</td> </tr> <tr> <td align=left><strong>neurology</strong></td> <td align=left><a id=fsl><code>fsl</code></a></td> <td align=left><code>5.0.10</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://fsl.fmrib.ox.ac.uk/fsl>Website</a></td> <td align=left>FSL is a comprehensive library of analysis tools for FMRI, MRI and DTI brain imaging data.</td> </tr> <tr> <td align=left><strong>neurology</strong></td> <td align=left><a id=mricron><code>mricron</code></a></td> <td align=left><code>20160502</code><br></td> <td align=left><a href=https://www.nitrc.org/projects/mricron>Website</a></td> <td align=left>MRIcron is a cross-platform NIfTI format image viewer.</td> </tr> <tr> <td align=left><strong>neurology</strong></td> <td align=left><a id=mrtrix><code>mrtrix</code></a></td> <td align=left><code>0.3.16</code><br></td> <td align=left><a href=http://www.mrtrix.org>Website</a></td> <td align=left>MRtrix3 provides a set of tools to perform various types of diffusion MRI analyses, from various forms of tractography through to next-generation group-level analyses.</td> </tr> <tr> <td align=left><strong>neurology</strong></td> <td align=left><a id=py-mdt><code>py-mdt</code></a></td> <td align=left><code>0.10.9_py36</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://github.com/cbclab/MDT>Website</a></td> <td align=left>The Maastricht Diffusion Toolbox, MDT, is a framework and library for parallelized (<abbr title="Graphical Processing Unit">GPU</abbr> and multi-core <abbr title="Central Processing Unit">CPU</abbr>) diffusion Magnetic Resonance Imaging (MRI) modeling.</td> </tr> <tr> <td align=left><strong>neurology</strong></td> <td align=left><a id=py-nipype><code>py-nipype</code></a></td> <td align=left><code>1.1.3_py27</code><br><code>1.1.3_py36</code><br></td> <td align=left><a href=https://nipype.readthedocs.io/en/latest/index.html>Website</a></td> <td align=left>Nipype is a Python project that provides a uniform interface to existing neuroimaging software and facilitates interaction between these packages within a single workflow.</td> </tr> <tr> <td align=left><strong>neurology</strong></td> <td align=left><a id=spm><code>spm</code></a></td> <td align=left><code>12</code><br></td> <td align=left><a href=http://www.fil.ion.ucl.ac.uk/spm/ >Website</a></td> <td align=left>The SPM software package has been designed for the analysis of brain imaging data sequences. The sequences can be a series of images from different cohorts, or time-series from the same subject.</td> </tr> <tr> <td align=left><strong>neurology</strong></td> <td align=left><a id=workbench><code>workbench</code></a></td> <td align=left><code>1.3.1</code><br></td> <td align=left><a href=https://www.humanconnectome.org/software/connectome-workbench>Website</a></td> <td align=left>Connectome Workbench is an open source, freely available visualization and discovery tool used to map neuroimaging data, especially data generated by the Human Connectome Project.</td> </tr> <tr> <td align=left><strong>pathology</strong></td> <td align=left><a id=openslide><code>openslide</code></a></td> <td align=left><code>3.4.1</code><br></td> <td align=left><a href=https://openslide.org/ >Website</a></td> <td align=left>OpenSlide is a C library that provides a simple interface to read whole-slide images (also known as virtual slides).</td> </tr> <tr> <td align=left><strong>pathology</strong></td> <td align=left><a id=py-openslide-python><code>py-openslide-python</code></a></td> <td align=left><code>1.1.1_py27</code><sup><b class=sw_def></b></sup><br><code>1.1.1_py36</code><br></td> <td align=left><a href=https://openslide.org/api/python/ >Website</a></td> <td align=left>OpenSlide Python is a Python interface to the OpenSlide library.</td> </tr> <tr> <td align=left><strong>phylogenetics</strong></td> <td align=left><a id=py-ete><code>py-ete</code></a></td> <td align=left><code>3.0.0_py27</code><br></td> <td align=left><a href=http://etetoolkit.org/ >Website</a></td> <td align=left>A Python framework for the analysis and visualization of trees.</td> </tr> <tr> <td align=left><strong>population genetics</strong></td> <td align=left><a id=py-admixfrog><code>py-admixfrog</code></a></td> <td align=left><code>0.6.1_py36</code><br></td> <td align=left><a href=https://github.com/BenjaminPeter/admixfrog>Website</a></td> <td align=left>Admixfrog is a HMM to infer ancestry frogments (fragments) from low-coverage, contaminated data.</td> </tr> </tbody> </table> <h3 id=chemistry><strong>chemistry</strong><a class=headerlink href=#chemistry title="Permanent link">#</a></h3> <table> <thead> <tr> <th align=left>Field</th> <th align=left><img style=float:left;min-width:110px;visibility:hidden>Module&nbsp;name</th> <th align=left><img style=float:left;min-width:90px;visibility:hidden>Version(s)</th> <th align=left>URL</th> <th align=left>Description</th> </tr> </thead> <tbody> <tr> <td align=left><strong>cheminformatics</strong></td> <td align=left><a id=py-rdkit><code>py-rdkit</code></a></td> <td align=left><code>2018.09.1_py27</code><sup><b class=sw_def></b></sup><br><code>2018.09.1_py36</code><br></td> <td align=left><a href=http://www.rdkit.org>Website</a></td> <td align=left>RDKit is a collection of cheminformatics and machine-learning software written in C++ and Python.</td> </tr> <tr> <td align=left><strong>computational chemistry</strong></td> <td align=left><a id=gaussian><code>gaussian</code></a></td> <td align=left><code>g16.A03</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_lic></b></sup><br><code>g16.B01</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_lic></b></sup><br></td> <td align=left><a href=https://www.gaussian.com>Website</a></td> <td align=left>Gaussian is a general purpose computational chemistry software package.</td> </tr> <tr> <td align=left><strong>computational chemistry</strong></td> <td align=left><a id=libint><code>libint</code></a></td> <td align=left><code>1.1.4</code><br><code>2.0.3</code><br></td> <td align=left><a href=https://sourceforge.net/p/libint/home/ >Website</a></td> <td align=left>Libint computes molecular integrals.</td> </tr> <tr> <td align=left><strong>computational chemistry</strong></td> <td align=left><a id=libxc><code>libxc</code></a></td> <td align=left><code>3.0.0</code><br></td> <td align=left><a href=http://octopus-code.org/wiki/Libxc>Website</a></td> <td align=left>Libxc is a library of exchange-correlation functionals for density-functional theory.</td> </tr> <tr> <td align=left><strong>computational chemistry</strong></td> <td align=left><a id=nwchem><code>nwchem</code></a></td> <td align=left><code>6.8</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=http://www.nwchem-sw.org>Website</a></td> <td align=left>NWChem is an ab initio computational chemistry software package which also includes quantum chemical and molecular dynamics functionality.</td> </tr> <tr> <td align=left><strong>computational chemistry</strong></td> <td align=left><a id=py-ase><code>py-ase</code></a></td> <td align=left><code>3.14.1_py27</code><br></td> <td align=left><a href=https://wiki.fysik.dtu.dk/ase/ >Website</a></td> <td align=left>The Atomic Simulation Environment (ASE) is a set of tools and Python modules for setting up, manipulating, running, visualizing and analyzing atomistic simulations.</td> </tr> <tr> <td align=left><strong>computational chemistry</strong></td> <td align=left><a id=schrodinger><code>schrodinger</code></a></td> <td align=left><code>2017-3</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><sup><b class=sw_lic></b></sup><br><code>2018-1</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><sup><b class=sw_lic></b></sup><br><code>2018-2</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><sup><b class=sw_lic></b></sup><br><code>2019-2</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><sup><b class=sw_lic></b></sup><br><code>2020-2</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><sup><b class=sw_lic></b></sup><br><code>2021-1</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><sup><b class=sw_lic></b></sup><br></td> <td align=left><a href=https://www.schrodinger.com>Website</a></td> <td align=left>Schrödinger Suites (Small-molecule Drug Discovery Suite, Material Science Suite, Biologics Suite) provide a set of molecular modelling software.</td> </tr> <tr> <td align=left><strong>computational chemistry</strong></td> <td align=left><a id=vasp><code>vasp</code></a></td> <td align=left><code>5.4.1</code><sup><b class=sw_def></b></sup><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><sup><b class=sw_lic></b></sup><br><code>6.6.1</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><sup><b class=sw_lic></b></sup><br></td> <td align=left><a href=https://www.vasp.at>Website</a></td> <td align=left>The Vienna Ab initio Simulation Package (VASP) is a computer program for atomic scale materials modelling, e.g. electronic structure calculations and quantum-mechanical molecular dynamics, from first principles.</td> </tr> <tr> <td align=left><strong>crystallography</strong></td> <td align=left><a id=vesta><code>vesta</code></a></td> <td align=left><code>3.4.4</code><br></td> <td align=left><a href=http://jp-minerals.org/vesta/en/ >Website</a></td> <td align=left>VESTA is a 3D visualization program for structural models, volumetric data such as electron/nuclear densities, and crystal morphologies.</td> </tr> <tr> <td align=left><strong>electrostatics</strong></td> <td align=left><a id=apbs><code>apbs</code></a></td> <td align=left><code>1.5</code><br></td> <td align=left><a href=http://www.poissonboltzmann.org/ >Website</a></td> <td align=left>APBS solves the equations of continuum electrostatics for large biomolecular assemblages.</td> </tr> <tr> <td align=left><strong>molecular dynamics</strong></td> <td align=left><a id=gromacs><code>gromacs</code></a></td> <td align=left><code>2016.3</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br><code>2018</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=http://www.gromacs.org>Website</a></td> <td align=left>GROMACS is a versatile package to perform molecular dynamics, i.e. simulate the Newtonian equations of motion for systems with hundreds to millions of particles.</td> </tr> <tr> <td align=left><strong>molecular dynamics</strong></td> <td align=left><a id=lammps><code>lammps</code></a></td> <td align=left><code>20180316</code><sup><b class=sw_mpi></b></sup><br><code>20200303</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=http://lammps.sandia.gov>Website</a></td> <td align=left>LAMMPS is a classical molecular dynamics code that models an ensemble of particles in a liquid, solid, or gaseous state.</td> </tr> <tr> <td align=left><strong>molecular dynamics</strong></td> <td align=left><a id=openmm><code>openmm</code></a></td> <td align=left><code>7.1.1</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=http://openmm.org>Website</a></td> <td align=left>A high performance toolkit for molecular simulation.</td> </tr> <tr> <td align=left><strong>molecular dynamics</strong></td> <td align=left><a id=plumed><code>plumed</code></a></td> <td align=left><code>2.3.2</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=http://www.plumed.org/ >Website</a></td> <td align=left>PLUMED is an open source library for free energy calculations in molecular systems.</td> </tr> <tr> <td align=left><strong>molecular dynamics</strong></td> <td align=left><a id=py-raspa2><code>py-raspa2</code></a></td> <td align=left><code>2.0.3_py27</code><br></td> <td align=left><a href=https://github.com/numat/RASPA2>Website</a></td> <td align=left>RASPA2 is a general purpose classical simulation package that can be used for the simulation of molecules in gases, fluids, zeolites, aluminosilicates, metal-organic frameworks, carbon nanotubes and external fields.</td> </tr> <tr> <td align=left><strong>molecular dynamics</strong></td> <td align=left><a id=qbox><code>qbox</code></a></td> <td align=left><code>1.65.0</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=http://www.qboxcode.org>Website</a></td> <td align=left>Qbox is a First-Principles Molecular Dynamics code.</td> </tr> <tr> <td align=left><strong>molecular dynamics</strong></td> <td align=left><a id=quip><code>quip</code></a></td> <td align=left><code>20170901</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://github.com/libAtoms/QUIP>Website</a></td> <td align=left>The QUIP package is a collection of software tools to carry out molecular dynamics simulations.</td> </tr> <tr> <td align=left><strong>quantum chemistry</strong></td> <td align=left><a id=cp2k><code>cp2k</code></a></td> <td align=left><code>4.1</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=http://www.cp2k.org>Website</a></td> <td align=left>CP2K is a quantum chemistry and solid state physics software package that can perform atomistic simulations of solid state, liquid, molecular, periodic, material, crystal, and biological systems.</td> </tr> <tr> <td align=left><strong>quantum chemistry</strong></td> <td align=left><a id=orca><code>orca</code></a></td> <td align=left><code>4.2.1</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://orcaforum.kofo.mpg.de/ >Website</a></td> <td align=left>ORCA is a flexible, efficient and easy-to-use general purpose tool for quantum chemistry.</td> </tr> <tr> <td align=left><strong>quantum chemistry</strong></td> <td align=left><a id=quantum-espresso><code>quantum-espresso</code></a></td> <td align=left><code>6.2.1</code><sup><b class=sw_mpi></b></sup><br><code>6.6</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://www.quantum-espresso.org>Website</a></td> <td align=left>Quantum ESPRESSO is an integrated suite of Open-Source computer codes for electronic-structure calculations and materials modeling at the nanoscale. It is based on density-functional theory, plane waves, and pseudopotentials.</td> </tr> <tr> <td align=left><strong>quantum chemistry</strong></td> <td align=left><a id=quantum-espresso_gpu><code>quantum-espresso_gpu</code></a></td> <td align=left><code>1.1</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://github.com/fspiga/qe-gpu>Website</a></td> <td align=left><abbr title="Graphical Processing Unit">GPU</abbr>-accelerated Quantum ESPRESSO using CUDA FORTRAN</td> </tr> </tbody> </table> <h3 id=devel><strong>devel</strong><a class=headerlink href=#devel title="Permanent link">#</a></h3> <table> <thead> <tr> <th align=left>Field</th> <th align=left><img style=float:left;min-width:110px;visibility:hidden>Module&nbsp;name</th> <th align=left><img style=float:left;min-width:90px;visibility:hidden>Version(s)</th> <th align=left>URL</th> <th align=left>Description</th> </tr> </thead> <tbody> <tr> <td align=left><strong>build</strong></td> <td align=left><a id=bazel><code>bazel</code></a></td> <td align=left><code>0.16.1</code><br><code>0.26.1</code><br><code>0.29.1</code><br></td> <td align=left><a href=https://bazel.build/ >Website</a></td> <td align=left>Bazel is a fast, scalable, multi-language and extensible build system.</td> </tr> <tr> <td align=left><strong>build</strong></td> <td align=left><a id=bazelisk><code>bazelisk</code></a></td> <td align=left><code>1.3.0</code><br><code>1.8.0</code><br></td> <td align=left><a href=https://github.com/bazelbuild/bazelisk>Website</a></td> <td align=left>Bazelisk is a wrapper for Bazel written in Go.</td> </tr> <tr> <td align=left><strong>build</strong></td> <td align=left><a id=cmake><code>cmake</code></a></td> <td align=left><code>3.8.1</code><br><code>3.11.1</code><br><code>3.13.1</code><br></td> <td align=left><a href=https://www.cmake.org>Website</a></td> <td align=left>CMake is an extensible, open-source system that manages the build process in an operating system and in a compiler-independent manner.</td> </tr> <tr> <td align=left><strong>build</strong></td> <td align=left><a id=kerl><code>kerl</code></a></td> <td align=left><code>1.8.5</code><br></td> <td align=left><a href=https://github.com/kerl/kerl>Website</a></td> <td align=left>Kerl is a tool to easily build and install Erlang/OTP instances.</td> </tr> <tr> <td align=left><strong>build</strong></td> <td align=left><a id=ninja><code>ninja</code></a></td> <td align=left><code>1.9.0</code><br></td> <td align=left><a href=https://ninja-build.org/ >Website</a></td> <td align=left>Ninja is a small build system with a focus on speed.</td> </tr> <tr> <td align=left><strong>build</strong></td> <td align=left><a id=py-meson><code>py-meson</code></a></td> <td align=left><code>0.51.1_py36</code><br></td> <td align=left><a href=https://mesonbuild.com/ >Website</a></td> <td align=left>Meson is an open source build system meant to be both extremely fast, and, even more importantly, as user friendly as possible.</td> </tr> <tr> <td align=left><strong>build</strong></td> <td align=left><a id=py-scons><code>py-scons</code></a></td> <td align=left><code>3.0.5_py27</code><sup><b class=sw_def></b></sup><br><code>3.0.5_py36</code><br></td> <td align=left><a href=http://www.scons.org>Website</a></td> <td align=left>SCons is an Open Source software construction tool.</td> </tr> <tr> <td align=left><strong>compiler</strong></td> <td align=left><a id=aocc><code>aocc</code></a></td> <td align=left><code>2.1.0</code><br><code>2.2.0</code><br></td> <td align=left><a href=https://developer.amd.com/amd-aocc>Website</a></td> <td align=left>AMD Optimizing C/C++ Compiler - AOCC is a highly optimized C, C++ and Fortran compiler for x86 targets especially for Zen based AMD processors.</td> </tr> <tr> <td align=left><strong>compiler</strong></td> <td align=left><a id=gcc><code>gcc</code></a></td> <td align=left><code>6.3.0</code><sup><b class=sw_def></b></sup><br><code>7.1.0</code><br><code>7.3.0</code><br><code>8.1.0</code><br><code>9.1.0</code><br><code>10.1.0</code><br></td> <td align=left><a href=http://gcc.gnu.org>Website</a></td> <td align=left>The GNU Compiler Collection includes front ends for C, C++, Fortran, Java, and Go, as well as libraries for these languages (libstdc++, libgcj,...).</td> </tr> <tr> <td align=left><strong>compiler</strong></td> <td align=left><a id=icc><code>icc</code></a></td> <td align=left><code>2017.u2</code><br><code>2018.u1</code><br><code>2018</code><br><code>2019</code><br></td> <td align=left><a href=https://software.intel.com/en-us/c-compilers>Website</a></td> <td align=left>Intel C++ Compiler, also known as icc or icl, is a group of C and C++ compilers from Intel</td> </tr> <tr> <td align=left><strong>compiler</strong></td> <td align=left><a id=ifort><code>ifort</code></a></td> <td align=left><code>2017.u2</code><br><code>2018.u1</code><br><code>2018</code><br><code>2019</code><br></td> <td align=left><a href=https://software.intel.com/en-us/fortran-compilers>Website</a></td> <td align=left>Intel Fortran Compiler, also known as ifort, is a group of Fortran compilers from Intel</td> </tr> <tr> <td align=left><strong>compiler</strong></td> <td align=left><a id=llvm><code>llvm</code></a></td> <td align=left><code>3.8.1</code><br><code>4.0.0</code><br><code>5.0.0</code><br><code>7.0.0</code><br></td> <td align=left><a href=http://llvm.org>Website</a></td> <td align=left>The LLVM Project is a collection of modular and reusable compiler and toolchain technologies. Clang is an LLVM native C/C++/Objective-C compiler,</td> </tr> <tr> <td align=left><strong>compiler</strong></td> <td align=left><a id=nagfor><code>nagfor</code></a></td> <td align=left><code>npl6a61na</code><br></td> <td align=left><a href=https://www.nag.com/nag-compiler>Website</a></td> <td align=left>The NAG Fortran Compiler is a full standard implementation of the ISO Fortran 95 language with the addition of all of Fortran 2003, most of Fortran 2008 and OpenMP 3.0 and 3.1.</td> </tr> <tr> <td align=left><strong>compiler</strong></td> <td align=left><a id=pgi><code>pgi</code></a></td> <td align=left><code>19.10</code><br></td> <td align=left><a href=http://pgroup.com>Website</a></td> <td align=left>PGI compilers and tools, including Open <abbr title="Message Passing Interface">MPI</abbr> (Community Edition).</td> </tr> <tr> <td align=left><strong>compiler</strong></td> <td align=left><a id=smlnj><code>smlnj</code></a></td> <td align=left><code>110.81</code><br></td> <td align=left><a href=http://smlnj.org/ >Website</a></td> <td align=left>Standard ML of New Jersey (abbreviated SML/NJ) is a compiler for the Standard ML '97 programming language.</td> </tr> <tr> <td align=left><strong>data</strong></td> <td align=left><a id=h5utils><code>h5utils</code></a></td> <td align=left><code>1.12.1</code><br></td> <td align=left><a href=http://ab-initio.mit.edu/wiki/index.php/H5utils>Website</a></td> <td align=left>h5utils is a set of utilities for visualization and conversion of scientific data in the free, portable HDF5 format.</td> </tr> <tr> <td align=left><strong>data</strong></td> <td align=left><a id=hdf5><code>hdf5</code></a></td> <td align=left><code>1.10.6</code><sup><b class=sw_def></b></sup><sup><b class=sw_mpi></b></sup><br><code>1.10.0p1</code><br><code>1.10.2</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://www.hdfgroup.org/hdf5>Website</a></td> <td align=left>HDF5 is a data model, library, and file format for storing and managing data. It supports an unlimited variety of datatypes, and is designed for flexible and efficient <abbr title="Input Output">I/O</abbr> and for high volume and complex data.</td> </tr> <tr> <td align=left><strong>data</strong></td> <td align=left><a id=hiredis><code>hiredis</code></a></td> <td align=left><code>0.13.3</code><br></td> <td align=left><a href=https://github.com/redis/hiredis>Website</a></td> <td align=left>Hiredis is a minimalistic C client library for the Redis database.</td> </tr> <tr> <td align=left><strong>data</strong></td> <td align=left><a id=ncl><code>ncl</code></a></td> <td align=left><code>6.4.0</code><br><code>6.6.2</code><br></td> <td align=left><a href=http://www.ncl.ucar.edu/ >Website</a></td> <td align=left>NCL is a free interpreted language designed specifically for scientific data processing and visualization.</td> </tr> <tr> <td align=left><strong>data</strong></td> <td align=left><a id=nco><code>nco</code></a></td> <td align=left><code>4.8.0</code><br></td> <td align=left><a href=http://nco.sourceforge.net/ >Website</a></td> <td align=left>The NCO toolkit manipulates and analyzes data stored in netCDF-accessible formats.</td> </tr> <tr> <td align=left><strong>data</strong></td> <td align=left><a id=netcdf><code>netcdf</code></a></td> <td align=left><code>4.4.1.1</code><br></td> <td align=left><a href=https://www.unidata.ucar.edu/software/netcdf>Website</a></td> <td align=left>NetCDF is a set of software libraries and self-describing, machine-independent data formats that support the creation, access, and sharing of array-oriented scientific data.</td> </tr> <tr> <td align=left><strong>data</strong></td> <td align=left><a id=pnetcdf><code>pnetcdf</code></a></td> <td align=left><code>1.8.1</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=http://cucis.ece.northwestern.edu/projects/PnetCDF/ >Website</a></td> <td align=left>Parallel netCDF (PnetCDF) is a parallel <abbr title="Input Output">I/O</abbr> library for accessing NetCDF files in CDF-1, 2, and 5 formats.</td> </tr> <tr> <td align=left><strong>data</strong></td> <td align=left><a id=protobuf><code>protobuf</code></a></td> <td align=left><code>3.4.0</code><br></td> <td align=left><a href=https://developers.google.com/protocol-buffers/ >Website</a></td> <td align=left>Protocol Buffers (a.k.a., protobuf) are Google's language-neutral, platform-neutral, extensible mechanism for serializing structured data.</td> </tr> <tr> <td align=left><strong>data</strong></td> <td align=left><a id=py-pandas><code>py-pandas</code></a></td> <td align=left><code>0.23.0_py27</code><br><code>0.23.0_py36</code><br><code>1.0.3_py36</code><br></td> <td align=left><a href=https://pandas.pydata.org/ >Website</a></td> <td align=left>pandas is an open source, BSD-licensed library providing high-performance, easy-to-use data structures and data analysis tools for the Python programming language.</td> </tr> <tr> <td align=left><strong>data</strong></td> <td align=left><a id=py-protobuf><code>py-protobuf</code></a></td> <td align=left><code>3.4.0_py27</code><sup><b class=sw_def></b></sup><br><code>3.4.0_py36</code><br><code>3.6.1_py27</code><br><code>3.6.1_py36</code><br><code>3.15.8_py36</code><br></td> <td align=left><a href=https://developers.google.com/protocol-buffers>Website</a></td> <td align=left>Python bindings for Google's Protocol Buffers data interchange format.</td> </tr> <tr> <td align=left><strong>data</strong></td> <td align=left><a id=redis><code>redis</code></a></td> <td align=left><code>4.0.1</code><br></td> <td align=left><a href=https://redis.io>Website</a></td> <td align=left>Redis is an open source, in-memory data structure store, used as a database, cache and message broker.</td> </tr> <tr> <td align=left><strong>data analytics</strong></td> <td align=left><a id=hadoop><code>hadoop</code></a></td> <td align=left><code>3.1.0</code><br></td> <td align=left><a href=https://hadoop.apache.org/ >Website</a></td> <td align=left>The Apache Hadoop software library is a framework that allows for the distributed processing of large data sets across clusters of computers using simple programming models.</td> </tr> <tr> <td align=left><strong>data analytics</strong></td> <td align=left><a id=py-sparkhpc><code>py-sparkhpc</code></a></td> <td align=left><code>0.3</code><br></td> <td align=left><a href=https://sparkhpc.readthedocs.io>Website</a></td> <td align=left>Launching and controlling spark on <abbr title="High Performance Computing">HPC</abbr> clusters</td> </tr> <tr> <td align=left><strong>data analytics</strong></td> <td align=left><a id=spark><code>spark</code></a></td> <td align=left><code>2.3.0</code><br></td> <td align=left><a href=https://spark.apache.org>Website</a></td> <td align=left>Apache Spark™ is a unified analytics engine for large-scale data processing.</td> </tr> <tr> <td align=left><strong>debug</strong></td> <td align=left><a id=gdb><code>gdb</code></a></td> <td align=left><code>8.2.1</code><br></td> <td align=left><a href=https://www.gnu.org/software/gdb/ >Website</a></td> <td align=left>GDB is the GNU Project debugger.</td> </tr> <tr> <td align=left><strong>debug</strong></td> <td align=left><a id=valgrind><code>valgrind</code></a></td> <td align=left><code>3.14.0</code><br></td> <td align=left><a href=http://valgrind.org/ >Website</a></td> <td align=left>Valgrind is an instrumentation framework for building dynamic analysis tools.</td> </tr> <tr> <td align=left><strong>engine</strong></td> <td align=left><a id=v8><code>v8</code></a></td> <td align=left><code>8.4.371.22</code><br></td> <td align=left><a href=https://v8.dev>Website</a></td> <td align=left>V8 is Google’s open source high-performance JavaScript and WebAssembly engine, written in C++.</td> </tr> <tr> <td align=left><strong>framework</strong></td> <td align=left><a id=dotnet><code>dotnet</code></a></td> <td align=left><code>2.1.500</code><br></td> <td align=left><a href=https://dotnet.microsoft.com/ >Website</a></td> <td align=left>.NET is a free, cross-platform, open source developer platform for building many different types of applications.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=cuda><code>cuda</code></a></td> <td align=left><code>9.0.176</code><sup><b class=sw_def></b></sup><sup><b class=sw_gpu></b></sup><br><code>8.0.61</code><sup><b class=sw_gpu></b></sup><br><code>9.1.85</code><sup><b class=sw_gpu></b></sup><br><code>9.2.88</code><sup><b class=sw_gpu></b></sup><br><code>9.2.148</code><sup><b class=sw_gpu></b></sup><br><code>10.0.130</code><sup><b class=sw_gpu></b></sup><br><code>10.1.105</code><sup><b class=sw_gpu></b></sup><br><code>10.1.168</code><sup><b class=sw_gpu></b></sup><br><code>10.2.89</code><sup><b class=sw_gpu></b></sup><br><code>11.0.3</code><sup><b class=sw_gpu></b></sup><br><code>11.1.1</code><sup><b class=sw_gpu></b></sup><br><code>11.2.0</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://developer.nvidia.com/cuda-toolkit>Website</a></td> <td align=left>CUDA is a parallel computing platform and application programming interface (API) model created by Nvidia. It allows software developers and software engineers to use a CUDA-enabled graphics processing unit (<abbr title="Graphical Processing Unit">GPU</abbr>) for general purpose processing.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=erlang><code>erlang</code></a></td> <td align=left><code>21.3</code><br></td> <td align=left><a href=http://erlang.org/ >Website</a></td> <td align=left>Erlang is a programming language used to build massively scalable soft real-time systems with requirements on high availability.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=go><code>go</code></a></td> <td align=left><code>1.9</code><br><code>1.14</code><br></td> <td align=left><a href=http://golang.org>Website</a></td> <td align=left>Go is an open source programming language that makes it easy to build simple, reliable, and efficient software.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=guile><code>guile</code></a></td> <td align=left><code>2.0.11</code><br><code>2.2.2</code><br></td> <td align=left><a href=https://www.gnu.org/software/guile/ >Website</a></td> <td align=left>GNU Guile is the preferred extension system for the GNU Project, which features an implementation of the Scheme programming language.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=haskell><code>haskell</code></a></td> <td align=left><code>8.6.5</code><br></td> <td align=left><a href=https://www.haskell.org/ >Website</a></td> <td align=left>Haskell is a statically typed, purely functional programming language with type inference and lazy evaluation.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=java><code>java</code></a></td> <td align=left><code>1.8.0_131</code><br></td> <td align=left><a href=http://www.oracle.com/technetwork/java/index.html>Website</a></td> <td align=left>Java is a general-purpose computer programming language that is concurrent, class-based, object-oriented,[14] and specifically designed to have as few implementation dependencies as possible.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=julia><code>julia</code></a></td> <td align=left><code>1.0.0</code><br><code>1.1.0</code><br><code>1.2.0</code><br><code>1.3.1</code><br><code>1.4.0</code><br><code>1.5.1</code><br></td> <td align=left><a href=https://julialang.org>Website</a></td> <td align=left>Julia is a high-level, high-performance dynamic programming language for numerical computing.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=lua><code>lua</code></a></td> <td align=left><code>5.3.4</code><br></td> <td align=left><a href=https://www.lua.org>Website</a></td> <td align=left>Lua is a powerful, efficient, lightweight, embeddable scripting language. It supports procedural programming, object-oriented programming, functional programming, data-driven programming, and data description.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=luarocks><code>luarocks</code></a></td> <td align=left><code>2.4.3</code><br></td> <td align=left><a href=https://luarocks.org/ >Website</a></td> <td align=left>LuaRocks is the package manager for Lua modules.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=manticore><code>manticore</code></a></td> <td align=left><code>20180301</code><br></td> <td align=left><a href=http://manticore.cs.uchicago.edu/ >Website</a></td> <td align=left>Manticore is a high-level parallel programming language aimed at general-purpose applications running on multi-core processors.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=nodejs><code>nodejs</code></a></td> <td align=left><code>8.9.4</code><br><code>9.5.0</code><br></td> <td align=left><a href=https://nodejs.org/ >Website</a></td> <td align=left>Node.js is a JavaScript runtime built on Chrome's V8 JavaScript engine. It provides the npm package manager.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=perl><code>perl</code></a></td> <td align=left><code>5.26.0</code><br></td> <td align=left><a href=https://www.perl.org>Website</a></td> <td align=left>Perl 5 is a highly capable, feature-rich programming language with over 29 years of development.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=php><code>php</code></a></td> <td align=left><code>7.3.0</code><br></td> <td align=left><a href=http://www.php.net>Website</a></td> <td align=left>PHP (recursive acronym for PHP: Hypertext Preprocessor) is an open source general-purpose scripting language that is especially suited for web development.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=py-cython><code>py-cython</code></a></td> <td align=left><code>0.27.3_py27</code><br><code>0.27.3_py36</code><br><code>0.29.21_py36</code><br></td> <td align=left><a href=http://cython.org/ >Website</a></td> <td align=left>Cython is an optimising static compiler for both the Python programming language and the extended Cython programming language (based on Pyrex).</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=py-ipython><code>py-ipython</code></a></td> <td align=left><code>5.4.1_py27</code><sup><b class=sw_def></b></sup><br><code>6.1.0_py36</code><br></td> <td align=left><a href=https://ipython.org>Website</a></td> <td align=left>IPython is a command shell for interactive computing in multiple programming languages, originally developed for the Python programming language.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=py-jupyter><code>py-jupyter</code></a></td> <td align=left><code>1.0.0_py27</code><sup><b class=sw_def></b></sup><br><code>1.0.0_py36</code><br></td> <td align=left><a href=https://jupyter.org>Website</a></td> <td align=left>Jupyter is a browser-based interactive notebook for programming, mathematics, and data science. It supports a number of languages via plugins.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=python><code>python</code></a></td> <td align=left><code>2.7.13</code><sup><b class=sw_def></b></sup><br><code>3.6.1</code><br><code>3.9.0</code><br></td> <td align=left><a href=https://www.python.org>Website</a></td> <td align=left>Python is an interpreted, interactive, object-oriented programming language.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=ruby><code>ruby</code></a></td> <td align=left><code>2.4.1</code><br><code>2.7.1</code><br></td> <td align=left><a href=https://www.ruby-lang.org>Website</a></td> <td align=left>A dynamic, open source programming language with a focus on simplicity and productivity. It has an elegant syntax that is natural to read and easy to write.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=rust><code>rust</code></a></td> <td align=left><code>1.35.0</code><br></td> <td align=left><a href=https://www.rust-lang.org/ >Website</a></td> <td align=left>A language empowering everyone to build reliable and efficient software.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=scala><code>scala</code></a></td> <td align=left><code>2.12.6</code><br></td> <td align=left><a href=https://www.scala-lang.org>Website</a></td> <td align=left>Scala combines object-oriented and functional programming in one concise, high-level language.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=ant><code>ant</code></a></td> <td align=left><code>1.10.1</code><br></td> <td align=left><a href=https://ant.apache.org/ >Website</a></td> <td align=left>Apache Ant is a Java library and command-line tool whose mission is to drive processes described in build files as targets and extension points dependent upon each other.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=boost><code>boost</code></a></td> <td align=left><code>1.64.0</code><br><code>1.69.0</code><br><code>1.75.0</code><br><code>1.76.0</code><br></td> <td align=left><a href=http://www.boost.org>Website</a></td> <td align=left>Boost is a set of libraries for the C++ programming language that provide support for tasks and structures such as linear algebra, pseudorandom number generation, multithreading, image processing, regular expressions, and unit testing.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=cnmem><code>cnmem</code></a></td> <td align=left><code>1.0.0</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://github.com/NVIDIA/ncmem>Website</a></td> <td align=left>CNMeM is a simple library to help the Deep Learning frameworks manage CUDA memory.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=cub><code>cub</code></a></td> <td align=left><code>1.7.3</code><sup><b class=sw_gpu></b></sup><br><code>1.10.0</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://nvlabs.github.io/cub/ >Website</a></td> <td align=left>CUB is a flexible library of cooperative threadblock primitives and other utilities for CUDA kernel programming.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=cutlass><code>cutlass</code></a></td> <td align=left><code>0.1.0</code><br></td> <td align=left><a href=https://github.com/NVIDIA/cutlass>Website</a></td> <td align=left>CUTLASS is a collection of CUDA C++ template abstractions for implementing high-performance matrix-multiplication (GEMM) at all levels and scales within CUDA.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=eigen><code>eigen</code></a></td> <td align=left><code>3.3.3</code><br></td> <td align=left><a href=http://eigen.tuxfamily.org>Website</a></td> <td align=left>Eigen is a C++ template library for linear algebra: matrices, vectors, numerical solvers, and related algorithms.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=libctl><code>libctl</code></a></td> <td align=left><code>3.2.2</code><br><code>4.0.1</code><br></td> <td align=left><a href=https://github.com/stevengj/libctl>Website</a></td> <td align=left>libctl is a library for supporting flexible control files in scientific simulations.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=libgpuarray><code>libgpuarray</code></a></td> <td align=left><code>0.7.5</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=http://deeplearning.net/software/libgpuarray>Website</a></td> <td align=left>Library to manipulate tensors on the <abbr title="Graphical Processing Unit">GPU</abbr>.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=nccl><code>nccl</code></a></td> <td align=left><code>1.3.4</code><sup><b class=sw_gpu></b></sup><br><code>2.0.4</code><sup><b class=sw_gpu></b></sup><br><code>2.1.15</code><sup><b class=sw_gpu></b></sup><br><code>2.2.13</code><sup><b class=sw_gpu></b></sup><br><code>2.3.7</code><sup><b class=sw_gpu></b></sup><br><code>2.4.8</code><sup><b class=sw_gpu></b></sup><br><code>2.5.6</code><sup><b class=sw_gpu></b></sup><br><code>2.8.4</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://github.com/NVIDIA/nccl>Website</a></td> <td align=left>NCCL (pronounced 'Nickel') is a stand-alone library of standard collective communication routines, such as all-gather, reduce, broadcast, etc., that have been optimized to achieve high bandwidth over PCIe.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=opencv><code>opencv</code></a></td> <td align=left><code>3.3.0</code><sup><b class=sw_def></b></sup><sup><b class=sw_gpu></b></sup><br><code>4.5.2</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://opencv.org>Website</a></td> <td align=left>OpenCV (Open Source Computer Vision Library) is an open source computer vision and machine learning software library.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=petsc><code>petsc</code></a></td> <td align=left><code>3.10.3</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://www.mcs.anl.gov/petsc/index.html>Website</a></td> <td align=left>PETSc is a suite of data structures and routines for the scalable (parallel) solution of scientific applications modeled by partial differential equations.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=py-h5py><code>py-h5py</code></a></td> <td align=left><code>2.7.1_py27</code><sup><b class=sw_def></b></sup><br><code>2.8.0_py36</code><br><code>2.10.0_py36</code><br></td> <td align=left><a href=http://www.h5py.org>Website</a></td> <td align=left>The h5py package is a Pythonic interface to the HDF5 binary data format.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=py-netcdf4><code>py-netcdf4</code></a></td> <td align=left><code>1.3.1_py27</code><sup><b class=sw_def></b></sup><br><code>1.3.1_py36</code><br></td> <td align=left><a href=https://unidata.github.io/netcdf4-python/ >Website</a></td> <td align=left>netcdf4-python is a Python interface to the netCDF C library.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=py-numba><code>py-numba</code></a></td> <td align=left><code>0.35.0_py27</code><sup><b class=sw_def></b></sup><br><code>0.35.0_py36</code><br><code>0.53.1_py36</code><br></td> <td align=left><a href=http://numba.pydata.org>Website</a></td> <td align=left>Numba is a compiler for Python array and numerical functions that gives you the power to speed up your applications with high performance functions written directly in Python..</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=py-pycuda><code>py-pycuda</code></a></td> <td align=left><code>2017.1.1_py27</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://mathema.tician.de/software/pycuda/ >Website</a></td> <td align=left>PyCUDA lets you access Nvidia‘s CUDA parallel computation API from Python.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=py-schwimmbad><code>py-schwimmbad</code></a></td> <td align=left><code>0.3.1_py36</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=http://scikit-image.org/ >Website</a></td> <td align=left>schwimmbad provides a uniform interface to parallel processing pools and enables switching easily between local development (e.g., serial processing or with multiprocessing) and deployment on a cluster or supercomputer (via, e.g., <abbr title="Message Passing Interface">MPI</abbr> or JobLib).</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=py-scikit-image><code>py-scikit-image</code></a></td> <td align=left><code>0.13.0_py27</code><br><code>0.14.0_py27</code><br><code>0.15.0_py27</code><br><code>0.15.0_py36</code><br><code>0.17.2_py36</code><br></td> <td align=left><a href=http://scikit-image.org/ >Website</a></td> <td align=left>scikit-image is a collection of algorithms for image processing.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=rabbitmq><code>rabbitmq</code></a></td> <td align=left><code>3.7.13</code><br></td> <td align=left><a href=http://rabbitmq.com>Website</a></td> <td align=left>RabbitMQ is an open-source message broker.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=swig><code>swig</code></a></td> <td align=left><code>3.0.12</code><br></td> <td align=left><a href=http://www.swig.org>Website</a></td> <td align=left>SWIG is an interface compiler that connects programs written in C and C++ with scripting languages such as Perl, Python, Ruby, and Tcl.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=tbb><code>tbb</code></a></td> <td align=left><code>2017.u2</code><br><code>2018.u1</code><br><code>2018</code><br><code>2019</code><br></td> <td align=left><a href=https://software.intel.com/en-us/intel-tbb>Website</a></td> <td align=left>Intel® Threading Building Blocks (Intel® TBB) is a widely used C++ library for shared-memory parallel programming and heterogeneous computing (intra-node distributed memory programming).</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=trilinos><code>trilinos</code></a></td> <td align=left><code>12.12.1</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://trilinos.org/ >Website</a></td> <td align=left>Trilinos is a collection of open-source software libraries, called packages, intended to be used as building blocks for the development of scientific applications.</td> </tr> <tr> <td align=left><strong>lib</strong></td> <td align=left><a id=zeromq><code>zeromq</code></a></td> <td align=left><code>4.2.2</code><br></td> <td align=left><a href=http://zeromq.org>Website</a></td> <td align=left>ZeroMQ (also spelled ØMQ, 0MQ or ZMQ) is a high-performance asynchronous messaging library, aimed at use in distributed or concurrent applications.</td> </tr> <tr> <td align=left><strong>mpi</strong></td> <td align=left><a id=hpcx><code>hpcx</code></a></td> <td align=left><code>2.6.0</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br><code>2.7.0</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br><code>2.8.1</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://www.mellanox.com/products/software/hpc-x>Website</a></td> <td align=left>Mellanox <abbr title="High Performance Computing">HPC</abbr>-X toolkit is a comprehensive software package that includes <abbr title="Message Passing Interface">MPI</abbr> and SHMEM/PGAS communications libraries.</td> </tr> <tr> <td align=left><strong>mpi</strong></td> <td align=left><a id=impi><code>impi</code></a></td> <td align=left><code>2017.u2</code><sup><b class=sw_mpi></b></sup><br><code>2018.u1</code><sup><b class=sw_mpi></b></sup><br><code>2018</code><sup><b class=sw_mpi></b></sup><br><code>2019</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://software.intel.com/en-us/intel-mpi-library>Website</a></td> <td align=left>Intel® <abbr title="Message Passing Interface">MPI</abbr> Library is a multi-fabric message passing library that implements the Message Passing Interface, version 3.1 (<abbr title="Message Passing Interface">MPI</abbr>-3.1) specification.</td> </tr> <tr> <td align=left><strong>mpi</strong></td> <td align=left><a id=openmpi><code>openmpi</code></a></td> <td align=left><code>2.0.2</code><sup><b class=sw_mpi></b></sup><br><code>2.1.1</code><sup><b class=sw_mpi></b></sup><br><code>3.1.2</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br><code>4.0.3</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br><code>4.0.5</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br><code>4.1.0</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://www.openmpi.org>Website</a></td> <td align=left>The Open <abbr title="Message Passing Interface">MPI</abbr> Project is an open source Message Passing Interface implementation that is developed and maintained by a consortium of academic, research, and industry partners.</td> </tr> <tr> <td align=left><strong>mpi</strong></td> <td align=left><a id=py-mpi4py><code>py-mpi4py</code></a></td> <td align=left><code>3.0.0_py27</code><sup><b class=sw_mpi></b></sup><br><code>3.0.3_py36</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://bitbucket.org/mpi4py/mpi4py>Website</a></td> <td align=left><abbr title="Message Passing Interface">MPI</abbr> for Python provides Python bindings for the Message Passing Interface (<abbr title="Message Passing Interface">MPI</abbr>) standard. It is implemented on top of the <abbr title="Message Passing Interface">MPI</abbr>-&frac12;/3 specification and exposes an API which grounds on the standard <abbr title="Message Passing Interface">MPI</abbr>-2 C++ bindings.</td> </tr> <tr> <td align=left><strong>networking</strong></td> <td align=left><a id=gasnet><code>gasnet</code></a></td> <td align=left><code>1.30.0</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://gasnet.lbl.gov>Website</a></td> <td align=left>GASNet is a language-independent, low-level networking layer that provides network-independent, high-performance communication primitives tailored for implementing parallel global address space SPMD languages and libraries.</td> </tr> <tr> <td align=left><strong>networking</strong></td> <td align=left><a id=libfabric><code>libfabric</code></a></td> <td align=left><code>1.6.0</code><br><code>1.6.2</code><br><code>1.7.1</code><br><code>1.9.1</code><br><code>1.10.1</code><br><code>1.11.1</code><br></td> <td align=left><a href=https://ofiwg.github.io/libfabric/ >Website</a></td> <td align=left>The Open Fabrics Interfaces (OFI) is a framework focused on exporting fabric communication services to applications. Libfabric is the library that defines and exports the user-space API of OFI.</td> </tr> <tr> <td align=left><strong>networking</strong></td> <td align=left><a id=ucx><code>ucx</code></a></td> <td align=left><code>1.3.1</code><br><code>1.8.1</code><sup><b class=sw_gpu></b></sup><br><code>1.9.0</code><sup><b class=sw_gpu></b></sup><br><code>1.10.0</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=http://www.openucx.org>Website</a></td> <td align=left>UCX is a communication library implementing high-performance messaging for <abbr title="Message Passing Interface">MPI</abbr>/PGAS frameworks.</td> </tr> <tr> <td align=left><strong>parser</strong></td> <td align=left><a id=xerces-c><code>xerces-c</code></a></td> <td align=left><code>3.2.1</code><br></td> <td align=left><a href=https://xerces.apache.org/xerces-c/index.html>Website</a></td> <td align=left>Xerces-C++ is a validating XML parser written in a portable subset of C++.</td> </tr> <tr> <td align=left><strong>profiling</strong></td> <td align=left><a id=amd-uprof><code>amd-uprof</code></a></td> <td align=left><code>3.3.462</code><br></td> <td align=left><a href=https://developer.amd.com/amd-uprof/ >Website</a></td> <td align=left>AMD uProf is a performance analysis tool for applications.</td> </tr> <tr> <td align=left><strong>runtime</strong></td> <td align=left><a id=starpu><code>starpu</code></a></td> <td align=left><code>1.3.2</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://gforge.inria.fr/projects/starpu/ >Website</a></td> <td align=left>StarPU is a unified runtime system that offers support for heterogeneous multicore architectures</td> </tr> <tr> <td align=left><strong>sdk</strong></td> <td align=left><a id=google-cloud-sdk><code>google-cloud-sdk</code></a></td> <td align=left><code>240.0.0</code><br><code>338.0.0</code><br></td> <td align=left><a href=https://cloud.google.com/sdk>Website</a></td> <td align=left>Command-line interface for Google Cloud Platform products and services.</td> </tr> </tbody> </table> <h3 id=math><strong>math</strong><a class=headerlink href=#math title="Permanent link">#</a></h3> <table> <thead> <tr> <th align=left>Field</th> <th align=left><img style=float:left;min-width:110px;visibility:hidden>Module&nbsp;name</th> <th align=left><img style=float:left;min-width:90px;visibility:hidden>Version(s)</th> <th align=left>URL</th> <th align=left>Description</th> </tr> </thead> <tbody> <tr> <td align=left><strong>computational geometry</strong></td> <td align=left><a id=cgal><code>cgal</code></a></td> <td align=left><code>4.10</code><br></td> <td align=left><a href=http://www.cgal.org>Website</a></td> <td align=left>The Computational Geometry Algorithms Library (CGAL) is a C++ library that aims to provide easy access to efficient and reliable algorithms in computational geometry.</td> </tr> <tr> <td align=left><strong>computational geometry</strong></td> <td align=left><a id=qhull><code>qhull</code></a></td> <td align=left><code>2015.2</code><br></td> <td align=left><a href=http://www.qhull.org/ >Website</a></td> <td align=left>Qhull computes the convex hull, Delaunay triangulation, Voronoi diagram, halfspace intersection about a point, furthest-site Delaunay triangulation, and furthest-site Voronoi diagram.</td> </tr> <tr> <td align=left><strong>deep learning</strong></td> <td align=left><a id=caffe2><code>caffe2</code></a></td> <td align=left><code>0.8.1</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://caffe2.ai>Website</a></td> <td align=left>Caffe2 is a deep learning framework that provides an easy and straightforward way to experiment with deep learning and leverage community contributions of new models and algorithms.</td> </tr> <tr> <td align=left><strong>deep learning</strong></td> <td align=left><a id=cudnn><code>cudnn</code></a></td> <td align=left><code>6.0</code><sup><b class=sw_gpu></b></sup><br><code>7.0.1</code><sup><b class=sw_gpu></b></sup><br><code>7.0.4</code><sup><b class=sw_gpu></b></sup><br><code>7.0.5</code><sup><b class=sw_gpu></b></sup><br><code>7.1.4</code><sup><b class=sw_gpu></b></sup><br><code>7.4.1.5</code><sup><b class=sw_gpu></b></sup><br><code>7.6.4</code><sup><b class=sw_gpu></b></sup><br><code>7.6.5</code><sup><b class=sw_gpu></b></sup><br><code>8.1.1.33</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://developer.nvidia.com/cudnn>Website</a></td> <td align=left>NVIDIA cuDNN is a <abbr title="Graphical Processing Unit">GPU</abbr>-accelerated library of primitives for deep neural networks.</td> </tr> <tr> <td align=left><strong>deep learning</strong></td> <td align=left><a id=cutensor><code>cutensor</code></a></td> <td align=left><code>1.2.0</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://developer.nvidia.com/cutensor>Website</a></td> <td align=left><abbr title="Graphical Processing Unit">GPU</abbr>-accelerated tensor linear algebra library.</td> </tr> <tr> <td align=left><strong>deep learning</strong></td> <td align=left><a id=py-horovod><code>py-horovod</code></a></td> <td align=left><code>0.12.1_py27</code><sup><b class=sw_def></b></sup><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br><code>0.12.1_py36</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://github.com/uber/horovod>Website</a></td> <td align=left>Horovod is a distributed training framework for TensorFlow. The goal of Horovod is to make distributed Deep Learning fast and easy to use.</td> </tr> <tr> <td align=left><strong>deep learning</strong></td> <td align=left><a id=py-keras><code>py-keras</code></a></td> <td align=left><code>2.1.5_py27</code><sup><b class=sw_def></b></sup><sup><b class=sw_gpu></b></sup><br><code>2.0.8_py27</code><sup><b class=sw_gpu></b></sup><br><code>2.1.5_py36</code><sup><b class=sw_gpu></b></sup><br><code>2.2.4_py27</code><sup><b class=sw_gpu></b></sup><br><code>2.2.4_py36</code><sup><b class=sw_gpu></b></sup><br><code>2.3.1_py36</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=http://keras.io>Website</a></td> <td align=left>Keras is a high-level neural networks API, written in Python and capable of running on top of TensorFlow, CNTK, or Theano.</td> </tr> <tr> <td align=left><strong>deep learning</strong></td> <td align=left><a id=py-onnx><code>py-onnx</code></a></td> <td align=left><code>1.0.1_py27</code><br><code>1.8.1_py36</code><br></td> <td align=left><a href=https://onnx.ai>Website</a></td> <td align=left>ONNX is a open format to represent deep learning models.</td> </tr> <tr> <td align=left><strong>deep learning</strong></td> <td align=left><a id=py-pytorch><code>py-pytorch</code></a></td> <td align=left><code>0.3.0_py27</code><sup><b class=sw_def></b></sup><sup><b class=sw_gpu></b></sup><br><code>0.2.0_py27</code><sup><b class=sw_gpu></b></sup><br><code>0.2.0_py36</code><sup><b class=sw_gpu></b></sup><br><code>0.3.0_py36</code><sup><b class=sw_gpu></b></sup><br><code>1.0.0_py27</code><sup><b class=sw_gpu></b></sup><br><code>1.0.0_py36</code><sup><b class=sw_gpu></b></sup><br><code>1.4.0_py36</code><sup><b class=sw_gpu></b></sup><br><code>1.8.1_py39</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=http://pytorch.org>Website</a></td> <td align=left>PyTorch is a deep learning framework that puts Python first.</td> </tr> <tr> <td align=left><strong>deep learning</strong></td> <td align=left><a id=py-tensorboardx><code>py-tensorboardx</code></a></td> <td align=left><code>1.8_py27</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://tensorboardx.readthedocs.io>Website</a></td> <td align=left>TensorboardX is TensorBoard™ for PyTorch (and Chainer, MXNet, NumPy...)</td> </tr> <tr> <td align=left><strong>deep learning</strong></td> <td align=left><a id=py-tensorflow><code>py-tensorflow</code></a></td> <td align=left><code>1.12.0_py27</code><sup><b class=sw_def></b></sup><sup><b class=sw_gpu></b></sup><br><code>1.4.0_py27</code><sup><b class=sw_gpu></b></sup><br><code>1.5.0_py27</code><sup><b class=sw_gpu></b></sup><br><code>1.5.0_py36</code><sup><b class=sw_gpu></b></sup><br><code>1.6.0_py27</code><sup><b class=sw_gpu></b></sup><br><code>1.6.0_py36</code><sup><b class=sw_gpu></b></sup><br><code>1.7.0_py27</code><sup><b class=sw_gpu></b></sup><br><code>1.8.0_py27</code><sup><b class=sw_gpu></b></sup><br><code>1.9.0_py27</code><sup><b class=sw_gpu></b></sup><br><code>1.9.0_py36</code><sup><b class=sw_gpu></b></sup><br><code>1.12.0_py36</code><sup><b class=sw_gpu></b></sup><br><code>2.0.0_py36</code><sup><b class=sw_gpu></b></sup><br><code>2.1.0_py36</code><sup><b class=sw_gpu></b></sup><br><code>2.4.1_py36</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://www.tensorflow.org>Website</a></td> <td align=left>TensorFlow™ is an open source software library for numerical computation using data flow graphs.</td> </tr> <tr> <td align=left><strong>deep learning</strong></td> <td align=left><a id=py-tensorlayer><code>py-tensorlayer</code></a></td> <td align=left><code>1.6.3_py27</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=http://tensorlayer.org>Website</a></td> <td align=left>TensorLayer is a Deep Learning (DL) and Reinforcement Learning (RL) library extended from Google TensorFlow.</td> </tr> <tr> <td align=left><strong>deep learning</strong></td> <td align=left><a id=py-theano><code>py-theano</code></a></td> <td align=left><code>1.0.1_py27</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=http://deeplearning.net/software/theano>Website</a></td> <td align=left>Theano is a Python library that allows you to define, optimize, and evaluate mathematical expressions involving multi-dimensional arrays efficiently.</td> </tr> <tr> <td align=left><strong>deep learning</strong></td> <td align=left><a id=tensorrt><code>tensorrt</code></a></td> <td align=left><code>3.0.1</code><sup><b class=sw_gpu></b></sup><br><code>3.0.4</code><sup><b class=sw_gpu></b></sup><br><code>4.0.1.6</code><sup><b class=sw_gpu></b></sup><br><code>5.0.2.6</code><sup><b class=sw_gpu></b></sup><br><code>6.0.1.8</code><sup><b class=sw_gpu></b></sup><br><code>7.0.0.11</code><sup><b class=sw_gpu></b></sup><br><code>7.2.3.4</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://developer.nvidia.com/tensorrt>Website</a></td> <td align=left>NVIDIA TensorRT™ is a high-performance deep learning inference optimizer and runtime that delivers low latency, high-throughput inference for deep learning applications.</td> </tr> <tr> <td align=left><strong>deep learning</strong></td> <td align=left><a id=torch><code>torch</code></a></td> <td align=left><code>20180202</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=http://torch.ch>Website</a></td> <td align=left>Torch is a scientific computing framework with wide support for machine learning algorithms that puts GPUs first.</td> </tr> <tr> <td align=left><strong>linear algebra</strong></td> <td align=left><a id=armadillo><code>armadillo</code></a></td> <td align=left><code>8.200.1</code><br></td> <td align=left><a href=http://arma.sourceforge.net/ >Website</a></td> <td align=left>Armadillo is a high quality linear algebra library (matrix maths) for the C++ language, aiming towards a good balance between speed and ease of use.</td> </tr> <tr> <td align=left><strong>machine learning</strong></td> <td align=left><a id=py-scikit-learn><code>py-scikit-learn</code></a></td> <td align=left><code>0.19.1_py27</code><sup><b class=sw_def></b></sup><br><code>0.19.1_py36</code><br><code>0.24.2_py36</code><br></td> <td align=left><a href=http://scikit-learn.org>Website</a></td> <td align=left>Scikit-learn is a free software machine learning library for the Python programming language.</td> </tr> <tr> <td align=left><strong>numerical analysis</strong></td> <td align=left><a id=matlab><code>matlab</code></a></td> <td align=left><code>R2017a</code><sup><b class=sw_lic></b></sup><br><code>R2017b</code><sup><b class=sw_lic></b></sup><br><code>R2018a</code><sup><b class=sw_lic></b></sup><br><code>R2019a</code><sup><b class=sw_lic></b></sup><br><code>R2020a</code><sup><b class=sw_lic></b></sup><br></td> <td align=left><a href=https://www.mathworks.com/products/matlab.html>Website</a></td> <td align=left>MATLAB is a multi-paradigm numerical computing environment and proprietary programming language developed by MathWorks.</td> </tr> <tr> <td align=left><strong>numerical analysis</strong></td> <td align=left><a id=octave><code>octave</code></a></td> <td align=left><code>4.2.1</code><br></td> <td align=left><a href=https://www.gnu.org/software/octave/ >Website</a></td> <td align=left>GNU Octave is a high-level language primarily intended for numerical computations.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=arpack><code>arpack</code></a></td> <td align=left><code>3.5.0</code><br><code>3.7.0</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://github.com/opencollab/arpack-ng>Website</a></td> <td align=left>Collection of Fortran77 subroutines designed to solve large scale eigenvalue problems.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=blis><code>blis</code></a></td> <td align=left><code>2.1</code><br><code>2.2.4</code><br></td> <td align=left><a href=https://github.com/amd/blis>Website</a></td> <td align=left>BLIS is a portable software framework for instantiating high-performance BLAS-like dense linear algebra libraries.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=fftw><code>fftw</code></a></td> <td align=left><code>3.3.6</code><sup><b class=sw_mpi></b></sup><br><code>3.3.8</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=http://www.fftw.org/ >Website</a></td> <td align=left>The Fastest Fourier Transform in the West (FFTW) is a software library for computing discrete Fourier transforms (DFTs).</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=glpk><code>glpk</code></a></td> <td align=left><code>4.63</code><br></td> <td align=left><a href=http://www.openblas.net/ >Website</a></td> <td align=left>The GLPK (GNU Linear Programming Kit) package is intended for solving large-scale linear programming (LP), mixed integer programming (MIP), and other related problems.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=gmp><code>gmp</code></a></td> <td align=left><code>6.1.2</code><br></td> <td align=left><a href=https://gmplib.org>Website</a></td> <td align=left>GMP is a free library for arbitrary precision arithmetic, operating on signed integers, rational numbers, and floating-point numbers.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=gsl><code>gsl</code></a></td> <td align=left><code>1.16</code><br><code>2.3</code><br></td> <td align=left><a href=https://www.gnu.org/software/gsl>Website</a></td> <td align=left>The GNU Scientific Library (GSL) is a numerical library for C and C++ programmers. The library provides a wide range of mathematical routines such as random number generators, special functions and least-squares fitting.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=harminv><code>harminv</code></a></td> <td align=left><code>1.4.1</code><br></td> <td align=left><a href=https://github.com/stevengj/harminv>Website</a></td> <td align=left>harminv is a program designed to solve the problem of harmonic inversion: given a time series consisting of a sum of sinusoids (modes), extract their frequencies and amplitudes.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=hypre><code>hypre</code></a></td> <td align=left><code>2.20.0</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://github.com/hypre-space/hypre>Website</a></td> <td align=left>HYPRE is a library of high performance preconditioners and solvers featuring multigrid methods for the solution of large, sparse linear systems of equations on massively parallel computers.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=imkl><code>imkl</code></a></td> <td align=left><code>2017.u2</code><br><code>2018.u1</code><br><code>2018</code><br><code>2019</code><br></td> <td align=left><a href=https://software.intel.com/en-us/intel-mkl>Website</a></td> <td align=left>Intel Math Kernel Library (Intel MKL) is a library of optimized math routines for science, engineering, and financial applications. Core math functions include BLAS, LAPACK, ScaLAPACK, sparse solvers, fast Fourier transforms, and vector math.[3] The routines in MKL are hand-optimized specifically for Intel processors</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=libflame><code>libflame</code></a></td> <td align=left><code>2.1</code><br><code>2.2.4</code><br></td> <td align=left><a href=https://github.com/amd/libflame>Website</a></td> <td align=left>libflame is a portable library for dense matrix computations, providing much of the functionality present in LAPACK</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=libxsmm><code>libxsmm</code></a></td> <td align=left><code>1.8.1</code><br></td> <td align=left><a href=https://github.com/hfp/libxsmm>Website</a></td> <td align=left>LIBXSMM is a library for small dense and small sparse matrix-matrix multiplications as well as for deep learning primitives such as small convolutions</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=metis><code>metis</code></a></td> <td align=left><code>5.1.0</code><br></td> <td align=left><a href=http://glaros.dtc.umn.edu/gkhome/metis/metis/overview>Website</a></td> <td align=left>METIS is a set of serial programs for partitioning graphs, partitioning finite element meshes, and producing fill reducing orderings for sparse matrices.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=mpc><code>mpc</code></a></td> <td align=left><code>1.2.1</code><br></td> <td align=left><a href=http://www.multiprecision.org/mpc>Website</a></td> <td align=left>GNU MPC is a C library for the arithmetic of complex numbers with arbitrarily high precision and correct rounding of the result.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=mpfr><code>mpfr</code></a></td> <td align=left><code>3.1.5</code><br><code>4.1.0</code><br></td> <td align=left><a href=https://www.mpfr.org>Website</a></td> <td align=left>The MPFR library is a C library for multiple-precision floating-point computations with correct rounding.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=mumps><code>mumps</code></a></td> <td align=left><code>5.1.2</code><br></td> <td align=left><a href=http://mumps.enseeiht.fr/ >Website</a></td> <td align=left>A parallel sparse direct solver.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=nagcl><code>nagcl</code></a></td> <td align=left><code>cll6i26dcl</code><br></td> <td align=left><a href=https://www.nag.com/nag-c-library>Website</a></td> <td align=left>The NAG C Library is the largest and most comprehensive collection of mathematical and statistical algorithms for C and C++.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=nagfl><code>nagfl</code></a></td> <td align=left><code>fll6i26dcl</code><br></td> <td align=left><a href=https://www.nag.com/nag-fortran-library>Website</a></td> <td align=left>The NAG Fortran Library is the largest and most comprehensive collection of numerical and statistical algorithms in Fortran.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=nagfs><code>nagfs</code></a></td> <td align=left><code>fsl6i26dcl</code><br></td> <td align=left><a href=https://www.nag.com/nag-library-smp-multicore>Website</a></td> <td align=left>The NAG Library for <abbr title="Shared Memory Parallelism">SMP</abbr> &amp; Multicore is based on, and includes, the full functionality of the NAG Fortran Library.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=nagmb><code>nagmb</code></a></td> <td align=left><code>MBL6I25DNL</code><br></td> <td align=left><a href=https://www.nag.com/nag-toolbox-matlab>Website</a></td> <td align=left>The NAG C Library is the largest and most comprehensive collection of mathematical and statistical algorithms for C and C++.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=openblas><code>openblas</code></a></td> <td align=left><code>0.3.4</code><sup><b class=sw_def></b></sup><br><code>0.2.19</code><br><code>0.3.9</code><br><code>0.3.10</code><br></td> <td align=left><a href=http://www.openblas.net/ >Website</a></td> <td align=left>OpenBLAS is an optimized BLAS library</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=parmetis><code>parmetis</code></a></td> <td align=left><code>4.0.3</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=http://glaros.dtc.umn.edu/gkhome/metis/parmetis/overview>Website</a></td> <td align=left>ParMETIS is an <abbr title="Message Passing Interface">MPI</abbr>-based parallel library that implements a variety of algorithms for partitioning unstructured graphs, meshes, and for computing fill-reducing orderings of sparse matrices.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=py-cupy><code>py-cupy</code></a></td> <td align=left><code>7.8.0_py36</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=http://cupy.dev>Website</a></td> <td align=left>CuPy is an implementation of NumPy-compatible multi-dimensional array on CUDA.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=py-gmpy2><code>py-gmpy2</code></a></td> <td align=left><code>2.0.8_py36</code><br></td> <td align=left><a href=https://gmpy2.readthedocs.io/ >Website</a></td> <td align=left>gmpy2 is a C-coded Python extension module that supports multiple-precision arithmetic.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=py-numpy><code>py-numpy</code></a></td> <td align=left><code>1.14.3_py27</code><sup><b class=sw_def></b></sup><br><code>1.14.3_py36</code><br><code>1.17.2_py36</code><br><code>1.18.1_py36</code><br><code>1.19.2_py36</code><br><code>1.20.3_py39</code><br></td> <td align=left><a href=http://www.numpy.org>Website</a></td> <td align=left>NumPy is the fundamental package for scientific computing with Python.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=py-pyublas><code>py-pyublas</code></a></td> <td align=left><code>2017.1_py27</code><br></td> <td align=left><a href=https://mathema.tician.de/software/pyublas/ >Website</a></td> <td align=left>PyUblas provides a seamless glue layer between Numpy and Boost.Ublas for use with Boost.Python.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=py-scipy><code>py-scipy</code></a></td> <td align=left><code>1.1.0_py27</code><sup><b class=sw_def></b></sup><br><code>1.1.0_py36</code><br><code>1.4.1_py36</code><br><code>1.6.3_py39</code><br></td> <td align=left><a href=https://www.scipy.org/scipylib>Website</a></td> <td align=left>The SciPy library provides many user-friendly and efficient numerical routines such as routines for numerical integration and optimization.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=qrupdate><code>qrupdate</code></a></td> <td align=left><code>1.1.2</code><br></td> <td align=left><a href=https://sourceforge.net/projects/qrupdate/ >Website</a></td> <td align=left>qrupdate is a Fortran library for fast updates of QR and Cholesky decompositions.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=scalapack><code>scalapack</code></a></td> <td align=left><code>2.0.2</code><sup><b class=sw_mpi></b></sup><br><code>2.1</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=http://www.netlib.org/scalapack/ >Website</a></td> <td align=left>ScaLAPACK is a library of high-performance linear algebra routines for parallel distributed memory machines.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=scotch><code>scotch</code></a></td> <td align=left><code>6.0.4</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=http://www.labri.fr/perso/pelegrin/scotch>Website</a></td> <td align=left>Software package and libraries for sequential and parallel graph partitioning, static mapping and clustering, sequential mesh and hypergraph partitioning, and sequential and parallel sparse matrix block ordering.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=superlu><code>superlu</code></a></td> <td align=left><code>5.2.1</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://github.com/xiaoyeli/superlu>Website</a></td> <td align=left>SuperLU is a general purpose library for the direct solution of large, sparse, nonsymmetric systems of linear equations.</td> </tr> <tr> <td align=left><strong>numerical library</strong></td> <td align=left><a id=xblas><code>xblas</code></a></td> <td align=left><code>1.0.248</code><br></td> <td align=left><a href=http://www.netlib.org/xblas>Website</a></td> <td align=left>Extra precise basic linear algebra subroutines.</td> </tr> <tr> <td align=left><strong>optimization</strong></td> <td align=left><a id=gurobi><code>gurobi</code></a></td> <td align=left><code>7.5.1</code><br><code>8.0.1_py27</code><br><code>8.0.1_py36</code><br><code>9.0.3_py36</code><br></td> <td align=left><a href=http://www.gurobi.com>Website</a></td> <td align=left>The Gurobi Optimizer is a commercial optimization solver for mathematical programming.</td> </tr> <tr> <td align=left><strong>optimization</strong></td> <td align=left><a id=knitro><code>knitro</code></a></td> <td align=left><code>10.3.0</code><br></td> <td align=left><a href=https://www.artelys.com/en/optimization-tools/knitro>Website</a></td> <td align=left>Artelys Knitro is an optimization solver for difficult large-scale nonlinear problems.</td> </tr> <tr> <td align=left><strong>optimization</strong></td> <td align=left><a id=nlopt><code>nlopt</code></a></td> <td align=left><code>2.6.2</code><br></td> <td align=left><a href=https://nlopt.readthedocs.io/ >Website</a></td> <td align=left>NLopt is a free/open-source library for nonlinear optimization.</td> </tr> <tr> <td align=left><strong>scientific computing</strong></td> <td align=left><a id=py-scipystack><code>py-scipystack</code></a></td> <td align=left><code>1.0_py27</code><sup><b class=sw_def></b></sup><br><code>1.0_py36</code><br></td> <td align=left><a href=https://www.scipy.org>Website</a></td> <td align=left>The SciPy Stack is a collection of open source software for scientific computing in Python. It provides the following packages: numpy, scipy, matplotlib, ipython, jupyter, pandas, sympy and nose.</td> </tr> <tr> <td align=left><strong>statistics</strong></td> <td align=left><a id=datamash><code>datamash</code></a></td> <td align=left><code>1.3</code><br></td> <td align=left><a href=https://www.gnu.org/software/datamash/ >Website</a></td> <td align=left>GNU datamash is a command-line program which performs basic numeric, textual and statistical operations on input textual data files.</td> </tr> <tr> <td align=left><strong>statistics</strong></td> <td align=left><a id=jags><code>jags</code></a></td> <td align=left><code>4.3.0</code><br></td> <td align=left><a href=http://mcmc-jags.sourceforge.net/ >Website</a></td> <td align=left>Just another Gibbs sampler (JAGS) is a program for simulation from Bayesian hierarchical models using Markov chain Monte Carlo (MCMC).</td> </tr> <tr> <td align=left><strong>statistics</strong></td> <td align=left><a id=py-rpy2><code>py-rpy2</code></a></td> <td align=left><code>2.8.6_py27</code><br><code>2.9.2_py36</code><br></td> <td align=left><a href=https://rpy2.bitbucket.io>Website</a></td> <td align=left>rpy2 is an interface to R running embedded in a Python process.</td> </tr> <tr> <td align=left><strong>statistics</strong></td> <td align=left><a id=R><code>R</code></a></td> <td align=left><code>3.5.1</code><sup><b class=sw_def></b></sup><br><code>3.4.0</code><br><code>3.6.1</code><br><code>4.0.2</code><br></td> <td align=left><a href=http://r-project.org>Website</a></td> <td align=left>R is a free software environment for statistical computing and graphics.</td> </tr> <tr> <td align=left><strong>statistics</strong></td> <td align=left><a id=rstudio><code>rstudio</code></a></td> <td align=left><code>1.1.423</code><br></td> <td align=left><a href=http://www.rstudio.com>Website</a></td> <td align=left>RStudio is an integrated development environment (IDE) for R. It includes a console, syntax-highlighting editor that supports direct code execution, as well as tools for plotting, history, debugging and workspace management.</td> </tr> <tr> <td align=left><strong>statistics</strong></td> <td align=left><a id=sas><code>sas</code></a></td> <td align=left><code>9.4</code><sup><b class=sw_lic></b></sup><br></td> <td align=left><a href=http://www.sas.com/ >Website</a></td> <td align=left>SAS is a software suite developed by SAS Institute for advanced analytics, multivariate analyses, business intelligence, data management, and predictive analytics.</td> </tr> <tr> <td align=left><strong>statistics</strong></td> <td align=left><a id=stata><code>stata</code></a></td> <td align=left><code>15</code><sup><b class=sw_def></b></sup><sup><b class=sw_lic></b></sup><br><code>14</code><sup><b class=sw_lic></b></sup><br><code>16</code><sup><b class=sw_lic></b></sup><br><code>17</code><sup><b class=sw_lic></b></sup><br></td> <td align=left><a href=https://www.stata.com>Website</a></td> <td align=left>Stata is a complete, integrated statistical software package that provides everything you need for data analysis, data management, and graphics.</td> </tr> <tr> <td align=left><strong>symbolic</strong></td> <td align=left><a id=libmatheval><code>libmatheval</code></a></td> <td align=left><code>1.1.11</code><br></td> <td align=left><a href=https://www.gnu.org/software/libmatheval/ >Website</a></td> <td align=left>GNU libmatheval is a library (callable from C and Fortran) to parse and evaluate symbolic expressions input as text.</td> </tr> <tr> <td align=left><strong>symbolic</strong></td> <td align=left><a id=py-sympy><code>py-sympy</code></a></td> <td align=left><code>1.1.1_py27</code><br><code>1.1.1_py36</code><br></td> <td align=left><a href=http://www.sympy.org>Website</a></td> <td align=left>SymPy is a Python library for symbolic mathematics.</td> </tr> </tbody> </table> <h3 id=physics><strong>physics</strong><a class=headerlink href=#physics title="Permanent link">#</a></h3> <table> <thead> <tr> <th align=left>Field</th> <th align=left><img style=float:left;min-width:110px;visibility:hidden>Module&nbsp;name</th> <th align=left><img style=float:left;min-width:90px;visibility:hidden>Version(s)</th> <th align=left>URL</th> <th align=left>Description</th> </tr> </thead> <tbody> <tr> <td align=left><strong>astronomy</strong></td> <td align=left><a id=heasoft><code>heasoft</code></a></td> <td align=left><code>6.22.1</code><br><code>6.26.1</code><br></td> <td align=left><a href=https://heasarc.gsfc.nasa.gov/docs/software/lheasoft/ >Website</a></td> <td align=left>HEAsoft is a Unified Release of the FTOOLS (General and mission-specific tools to manipulate FITS files) and XANADU (High-level, multi-mission tasks for X-ray astronomical spectral, timing, and imaging data analysis) software packages.</td> </tr> <tr> <td align=left><strong>astronomy</strong></td> <td align=left><a id=py-astropy><code>py-astropy</code></a></td> <td align=left><code>4.0.1_py36</code><br></td> <td align=left><a href=https://www.astropy.org/ >Website</a></td> <td align=left>The Astropy Project is a community effort to develop a common core package for Astronomy in Python and foster an ecosystem of interoperable astronomy packages.</td> </tr> <tr> <td align=left><strong>astronomy</strong></td> <td align=left><a id=py-lenstools><code>py-lenstools</code></a></td> <td align=left><code>1.0_py36</code><br></td> <td align=left><a href=https://lenstools.readthedocs.io/ >Website</a></td> <td align=left>This python package collects together a suite of widely used analysis tools in Weak Gravitational Lensing.</td> </tr> <tr> <td align=left><strong>CFD</strong></td> <td align=left><a id=su2><code>su2</code></a></td> <td align=left><code>7.0.3</code><br></td> <td align=left><a href=https://su2code.github.io/ >Website</a></td> <td align=left>SU2: An Open-Source Suite for Multiphysics Simulation and Design</td> </tr> <tr> <td align=left><strong>climate modeling</strong></td> <td align=left><a id=cdo><code>cdo</code></a></td> <td align=left><code>1.9.7.1</code><br></td> <td align=left><a href=https://code.mpimet.mpg.de/projects/cdo>Website</a></td> <td align=left>CDO is a collection of command line Operators to manipulate and analyse Climate and NWP model Data.</td> </tr> <tr> <td align=left><strong>geophysics</strong></td> <td align=left><a id=opensees><code>opensees</code></a></td> <td align=left><code>2.5.0</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=http://opensees.berkeley.edu/ >Website</a></td> <td align=left>OpenSees is a software framework for developing applications to simulate the performance of structural and geotechnical systems subjected to earthquakes.</td> </tr> <tr> <td align=left><strong>geoscience</strong></td> <td align=left><a id=gdal><code>gdal</code></a></td> <td align=left><code>2.2.1</code><br></td> <td align=left><a href=http://www.gdal.org>Website</a></td> <td align=left>GDAL is a translator library for raster and vector geospatial data formats.</td> </tr> <tr> <td align=left><strong>geoscience</strong></td> <td align=left><a id=geos><code>geos</code></a></td> <td align=left><code>3.6.2</code><br></td> <td align=left><a href=https://trac.osgeo.org/geos>Website</a></td> <td align=left>GEOS (Geometry Engine - Open Source) is a C++ port of Java Topology Suite (JTS).</td> </tr> <tr> <td align=left><strong>geoscience</strong></td> <td align=left><a id=proj><code>proj</code></a></td> <td align=left><code>4.9.3</code><br></td> <td align=left><a href=http://proj4.org>Website</a></td> <td align=left>proj.4 is a standard UNIX filter function which converts geographic longitude and latitude coordinates into cartesian coordinates (and vice versa.</td> </tr> <tr> <td align=left><strong>geoscience</strong></td> <td align=left><a id=py-opendrift><code>py-opendrift</code></a></td> <td align=left><code>1.0.3_py27</code><br></td> <td align=left><a href=https://github.com/OpenDrift/opendrift>Website</a></td> <td align=left>OpenDrift is a software for modeling the trajectories and fate of objects or substances drifting in the ocean, or even in the atmosphere.</td> </tr> <tr> <td align=left><strong>geoscience</strong></td> <td align=left><a id=py-pyproj><code>py-pyproj</code></a></td> <td align=left><code>1.9.5.1_py27</code><sup><b class=sw_def></b></sup><br><code>1.9.5.1_py36</code><br></td> <td align=left><a href=https://github.com/jswhit/pyproj>Website</a></td> <td align=left>Python interface to PROJ4 library for cartographic transformations.</td> </tr> <tr> <td align=left><strong>geoscience</strong></td> <td align=left><a id=udunits><code>udunits</code></a></td> <td align=left><code>2.2.26</code><br></td> <td align=left><a href=https://www.unidata.ucar.edu/software/udunits/ >Website</a></td> <td align=left>The UDUNITS package from Unidata is a C-based package for the programatic handling of units of physical quantities.</td> </tr> <tr> <td align=left><strong>materials science</strong></td> <td align=left><a id=atat><code>atat</code></a></td> <td align=left><code>3.36</code><br></td> <td align=left><a href=https://www.brown.edu/Departments/Engineering/Labs/avdw/atat/ >Website</a></td> <td align=left>Alloy Theoretic Automated Toolkit: a software toolkit for modeling coupled configurational and vibrational disorder in alloy systems.</td> </tr> <tr> <td align=left><strong>micromagnetics</strong></td> <td align=left><a id=oommf><code>oommf</code></a></td> <td align=left><code>1.2b4</code><br></td> <td align=left><a href=https://math.nist.gov/oommf>Website</a></td> <td align=left>OOMMF is a set of portable, extensible public domain micromagnetic program and associated tools.</td> </tr> <tr> <td align=left><strong>particle</strong></td> <td align=left><a id=openmc><code>openmc</code></a></td> <td align=left><code>0.10.0</code><br></td> <td align=left><a href=https://github.com/openmc-dev/openmc>Website</a></td> <td align=left>OpenMC is a Monte Carlo particle transport simulation code focused on neutron criticality calculations.</td> </tr> <tr> <td align=left><strong>photonics</strong></td> <td align=left><a id=meep><code>meep</code></a></td> <td align=left><code>1.3</code><sup><b class=sw_mpi></b></sup><br><code>1.4.3</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://github.com/stevengj/meep>Website</a></td> <td align=left>Meep is a free finite-difference time-domain (FDTD) simulation software package to model electromagnetic systems.</td> </tr> <tr> <td align=left><strong>photonics</strong></td> <td align=left><a id=mpb><code>mpb</code></a></td> <td align=left><code>1.5</code><sup><b class=sw_mpi></b></sup><br><code>1.6.2</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://github.com/stevengj/mpb>Website</a></td> <td align=left>MPB is a free software package for computing the band structures, or dispersion relations, and electromagnetic modes of periodic dielectric structures, on both serial and parallel computers.</td> </tr> <tr> <td align=left><strong>quantum mechanics</strong></td> <td align=left><a id=py-quspin><code>py-quspin</code></a></td> <td align=left><code>0.3.5_py36</code><br></td> <td align=left><a href=http://weinbe58.github.io/QuSpin/ >Website</a></td> <td align=left>QuSpin is an open-source Python package for exact diagonalization and quantum dynamics of arbitrary boson, fermion and spin many-body systems.</td> </tr> <tr> <td align=left><strong>quantum mechanics</strong></td> <td align=left><a id=py-qutip><code>py-qutip</code></a></td> <td align=left><code>4.5.2_py36</code><br></td> <td align=left><a href=http://qutip.org/ >Website</a></td> <td align=left>QuTiP is open-source software for simulating the dynamics of closed and open quantum systems.</td> </tr> </tbody> </table> <h3 id=system><strong>system</strong><a class=headerlink href=#system title="Permanent link">#</a></h3> <table> <thead> <tr> <th align=left>Field</th> <th align=left><img style=float:left;min-width:110px;visibility:hidden>Module&nbsp;name</th> <th align=left><img style=float:left;min-width:90px;visibility:hidden>Version(s)</th> <th align=left>URL</th> <th align=left>Description</th> </tr> </thead> <tbody> <tr> <td align=left><strong>scm</strong></td> <td align=left><a id=gh><code>gh</code></a></td> <td align=left><code>1.9.1</code><br></td> <td align=left><a href=https://cli.github.com/ >Website</a></td> <td align=left>gh is GitHub on the command line. It brings pull requests, issues, and other GitHub concepts to the terminal next to where you are already working with git and your code.</td> </tr> <tr> <td align=left><strong>backup</strong></td> <td align=left><a id=restic><code>restic</code></a></td> <td align=left><code>0.9.5</code><br><code>0.12.0</code><br></td> <td align=left><a href=https://restic.net>Website</a></td> <td align=left>Fast, secure, efficient backup program.</td> </tr> <tr> <td align=left><strong>benchmark</strong></td> <td align=left><a id=hp2p><code>hp2p</code></a></td> <td align=left><code>3.2</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://github.com/cea-hpc/hp2p>Website</a></td> <td align=left>Heavy Peer To Peer: a <abbr title="Message Passing Interface">MPI</abbr> based benchmark for network diagnostic.</td> </tr> <tr> <td align=left><strong>benchmark</strong></td> <td align=left><a id=mpibench><code>mpibench</code></a></td> <td align=left><code>20190729</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://github.com/LLNL/mpiBench>Website</a></td> <td align=left>Times <abbr title="Message Passing Interface">MPI</abbr> collectives over a series of message sizes.</td> </tr> <tr> <td align=left><strong>benchmark</strong></td> <td align=left><a id=mprime><code>mprime</code></a></td> <td align=left><code>29.4</code><br></td> <td align=left><a href=https://www.mersenne.org/ >Website</a></td> <td align=left>mprime is used by GIMPS, a distributed computing project dedicated to finding new Mersenne prime numbers, and which is commonly used as a stability testing utility.</td> </tr> <tr> <td align=left><strong>benchmark</strong></td> <td align=left><a id=osu-micro-benchmarks><code>osu-micro-benchmarks</code></a></td> <td align=left><code>5.6.1</code><sup><b class=sw_mpi></b></sup><br><code>5.6.3</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br><code>5.7</code><sup><b class=sw_gpu></b></sup><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=http://mvapich.cse.ohio-state.edu/benchmarks/ >Website</a></td> <td align=left>The OSU MicroBenchmarks carry out a variety of message passing performance tests using <abbr title="Message Passing Interface">MPI</abbr>.</td> </tr> <tr> <td align=left><strong>checkpointing</strong></td> <td align=left><a id=dmtcp><code>dmtcp</code></a></td> <td align=left><code>2.6.0</code><br></td> <td align=left><a href=http://dmtcp.sourceforge.net/ >Website</a></td> <td align=left>DMTCP (Distributed MultiThreaded Checkpointing) transparently checkpoints a single-host or distributed computation in user-space -- with no modifications to user code or to the O/S.</td> </tr> <tr> <td align=left><strong>compression</strong></td> <td align=left><a id=libarchive><code>libarchive</code></a></td> <td align=left><code>3.3.2</code><br><code>3.4.2</code><br></td> <td align=left><a href=https://www.libarchive.org/ >Website</a></td> <td align=left>The libarchive project develops a portable, efficient C library that can read and write streaming archives in a variety of formats.</td> </tr> <tr> <td align=left><strong>compression</strong></td> <td align=left><a id=libzip><code>libzip</code></a></td> <td align=left><code>1.5.1</code><br></td> <td align=left><a href=http://libzip.org>Website</a></td> <td align=left>libzip is a C library for reading, creating, and modifying zip archives.</td> </tr> <tr> <td align=left><strong>compression</strong></td> <td align=left><a id=lz4><code>lz4</code></a></td> <td align=left><code>1.8.0</code><br></td> <td align=left><a href=https://www.lz4.org>Website</a></td> <td align=left>LZ4 is lossless compression algorithm.</td> </tr> <tr> <td align=left><strong>compression</strong></td> <td align=left><a id=lzo><code>lzo</code></a></td> <td align=left><code>2.10</code><br></td> <td align=left><a href=http://www.oberhumer.com/opensource/lzo>Website</a></td> <td align=left>LZO is a portable lossless data compression library written in ANSI C.</td> </tr> <tr> <td align=left><strong>compression</strong></td> <td align=left><a id=mpibzip2><code>mpibzip2</code></a></td> <td align=left><code>0.6</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://compression.ca/mpibzip2>Website</a></td> <td align=left>MPIBZIP2 is a parallel implementation of the bzip2 block-sorting file compressor that uses <abbr title="Message Passing Interface">MPI</abbr> and achieves significant speedup on cluster machines.</td> </tr> <tr> <td align=left><strong>compression</strong></td> <td align=left><a id=p7zip><code>p7zip</code></a></td> <td align=left><code>16.02</code><br></td> <td align=left><a href=https://sourceforge.net/projects/p7zip/ >Website</a></td> <td align=left>p7zip is a Linux port of 7zip, a file archiver with high compression ratio.</td> </tr> <tr> <td align=left><strong>compression</strong></td> <td align=left><a id=pbzip2><code>pbzip2</code></a></td> <td align=left><code>1.1.12</code><br></td> <td align=left><a href=https://compression.ca/pbzip2>Website</a></td> <td align=left>PBZIP2 is a parallel implementation of the bzip2 block-sorting file compressor that uses pthreads and achieves near-linear speedup on <abbr title="Shared Memory Parallelism">SMP</abbr> machines.</td> </tr> <tr> <td align=left><strong>compression</strong></td> <td align=left><a id=pigz><code>pigz</code></a></td> <td align=left><code>2.4</code><br></td> <td align=left><a href=https://zlib.net/pigz/ >Website</a></td> <td align=left>A parallel implementation of gzip for modern multi-processor, multi-core machines.</td> </tr> <tr> <td align=left><strong>compression</strong></td> <td align=left><a id=szip><code>szip</code></a></td> <td align=left><code>2.1.1</code><br></td> <td align=left><a href=https://support.hdfgroup.org/doc_resource/SZIP>Website</a></td> <td align=left>Szip compression software, providing lossless compression of scientific data, is an implementation of the extended-Rice lossless compression algorithm.</td> </tr> <tr> <td align=left><strong>compression</strong></td> <td align=left><a id=xz><code>xz</code></a></td> <td align=left><code>5.2.3</code><br></td> <td align=left><a href=http://tukaani.org/xz/ >Website</a></td> <td align=left>XZ Utils is free general-purpose data compression software with a high compression ratio.</td> </tr> <tr> <td align=left><strong>compression</strong></td> <td align=left><a id=zlib><code>zlib</code></a></td> <td align=left><code>1.2.11</code><br></td> <td align=left><a href=http://zlib.net>Website</a></td> <td align=left>zlib is designed to be a free, general-purpose, legally unencumbered -- that is, not covered by any patents -- lossless data-compression library for use on virtually any computer hardware and operating system.</td> </tr> <tr> <td align=left><strong>containers</strong></td> <td align=left><a id=libnvidia-container><code>libnvidia-container</code></a></td> <td align=left><code>1.0.0rc2</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://github.com/NVIDIA/libnvidia-container>Website</a></td> <td align=left>libnvidia-container is a library and a simple CLI utility to automatically configure GNU/Linux containers leveraging NVIDIA hardware.</td> </tr> <tr> <td align=left><strong>containers</strong></td> <td align=left><a id=proot><code>proot</code></a></td> <td align=left><code>5.1.0</code><br></td> <td align=left><a href=https://proot-me.github.io>Website</a></td> <td align=left>PRoot is a user-space implementation of chroot, mount --bind, and binfmt_misc.</td> </tr> <tr> <td align=left><strong>database</strong></td> <td align=left><a id=bdb><code>bdb</code></a></td> <td align=left><code>6.2.32</code><br></td> <td align=left><a href=http://www.oracle.com/us/products/database/berkeley-db>Website</a></td> <td align=left>Berkeley DB (BDB) is a software library intended to provide a high-performance embedded database for key/value data.</td> </tr> <tr> <td align=left><strong>database</strong></td> <td align=left><a id=mariadb><code>mariadb</code></a></td> <td align=left><code>10.2.11</code><br></td> <td align=left><a href=https://www.mariadb.org>Website</a></td> <td align=left>MariaDB is a community-developed fork of the MySQL relational database management system intended to remain free under the GNU GPL.</td> </tr> <tr> <td align=left><strong>database</strong></td> <td align=left><a id=postgresql><code>postgresql</code></a></td> <td align=left><code>10.5</code><br></td> <td align=left><a href=https://postgresql.org>Website</a></td> <td align=left>PostgreSQL is a powerful, open source object-relational database system with a strong focus on reliability, feature robustness, and performance.</td> </tr> <tr> <td align=left><strong>database</strong></td> <td align=left><a id=sqlite><code>sqlite</code></a></td> <td align=left><code>3.18.0</code><br></td> <td align=left><a href=https://www.sqlite.org>Website</a></td> <td align=left>SQLite is a self-contained, high-reliability, embedded, full-featured, public-domain, SQL database engine.</td> </tr> <tr> <td align=left><strong>document management</strong></td> <td align=left><a id=pandoc><code>pandoc</code></a></td> <td align=left><code>2.7.3</code><br></td> <td align=left><a href=https://pandoc.org>Website</a></td> <td align=left>Pandoc is a universal document converter.</td> </tr> <tr> <td align=left><strong>document processing</strong></td> <td align=left><a id=ghostscript><code>ghostscript</code></a></td> <td align=left><code>9.53.2</code><br></td> <td align=left><a href=https://ghostcript.com>Website</a></td> <td align=left>Ghostscript is an interpreter for the PostScript language and PDF files.</td> </tr> <tr> <td align=left><strong>document processing</strong></td> <td align=left><a id=lyx><code>lyx</code></a></td> <td align=left><code>2.3.2</code><br></td> <td align=left><a href=http://www.lyx.org>Website</a></td> <td align=left>LyX is a document processor.</td> </tr> <tr> <td align=left><strong>document processing</strong></td> <td align=left><a id=poppler><code>poppler</code></a></td> <td align=left><code>0.47.0</code><br></td> <td align=left><a href=https://poppler.freedesktop.org/ >Website</a></td> <td align=left>Poppler is a PDF rendering library.</td> </tr> <tr> <td align=left><strong>document processing</strong></td> <td align=left><a id=texinfo><code>texinfo</code></a></td> <td align=left><code>6.6</code><br></td> <td align=left><a href=https://www.gnu.org/software/texinfo/ >Website</a></td> <td align=left>Texinfo is the official documentation format of the GNU project.</td> </tr> <tr> <td align=left><strong>document processing</strong></td> <td align=left><a id=texlive><code>texlive</code></a></td> <td align=left><code>2019</code><br></td> <td align=left><a href=https://www.tug.org/texlive/ >Website</a></td> <td align=left>TeX Live is an easy way to get up and running with the TeX document production system.</td> </tr> <tr> <td align=left><strong>file management</strong></td> <td align=left><a id=duc><code>duc</code></a></td> <td align=left><code>1.4.4</code><br></td> <td align=left><a href=https://duc.zevv.nl>Website</a></td> <td align=left>Duc is a collection of tools for indexing, inspecting and visualizing disk usage.</td> </tr> <tr> <td align=left><strong>file management</strong></td> <td align=left><a id=exa><code>exa</code></a></td> <td align=left><code>0.8.0</code><br></td> <td align=left><a href=https://github.com/ogham/exa>Website</a></td> <td align=left>exa is a replacement for ls written in Rust.</td> </tr> <tr> <td align=left><strong>file management</strong></td> <td align=left><a id=fpart><code>fpart</code></a></td> <td align=left><code>0.9.3</code><br></td> <td align=left><a href=https://github.com/martymac/fpart>Website</a></td> <td align=left>fpart sorts files and packs them into partitions.</td> </tr> <tr> <td align=left><strong>file management</strong></td> <td align=left><a id=ncdu><code>ncdu</code></a></td> <td align=left><code>1.15.1</code><br></td> <td align=left><a href=https://dev.yorhel.nl/ncdu>Website</a></td> <td align=left>Ncdu is a disk usage analyzer with an ncurses interface.</td> </tr> <tr> <td align=left><strong>file management</strong></td> <td align=left><a id=py-pcircle><code>py-pcircle</code></a></td> <td align=left><code>0.17_py27</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://github.com/olcf/pcircle>Website</a></td> <td align=left>pcircle contains a suite of file system tools developed at OLCF to take advantage of highly scalable parallel file system such as Lustre.</td> </tr> <tr> <td align=left><strong>file management</strong></td> <td align=left><a id=rmlint><code>rmlint</code></a></td> <td align=left><code>2.8.0</code><br></td> <td align=left><a href=https://rmlint.readthedocs.io/ >Website</a></td> <td align=left>rmlint finds space waste and other broken things on your filesystem and offers to remove it.</td> </tr> <tr> <td align=left><strong>file transfer</strong></td> <td align=left><a id=aria2><code>aria2</code></a></td> <td align=left><code>1.35.0</code><br></td> <td align=left><a href=https://aria2.github.io/ >Website</a></td> <td align=left>aria2 is a lightweight multi-protocol &amp; multi-source command-line download utility.</td> </tr> <tr> <td align=left><strong>file transfer</strong></td> <td align=left><a id=aspera-cli><code>aspera-cli</code></a></td> <td align=left><code>3.9.6</code><br></td> <td align=left><a href=https://www.ibm.com/support/knowledgecenter/SS4F2E>Website</a></td> <td align=left>The IBM Aspera Command-Line Interface (the Aspera CLI) is a collection of Aspera tools for performing high-speed, secure data transfers from the command line.</td> </tr> <tr> <td align=left><strong>file transfer</strong></td> <td align=left><a id=aws-cli><code>aws-cli</code></a></td> <td align=left><code>2.0.50</code><br></td> <td align=left><a href=https://github.com/aws/aws-cli>Website</a></td> <td align=left>This package provides a unified command line interface to Amazon Web Services.</td> </tr> <tr> <td align=left><strong>file transfer</strong></td> <td align=left><a id=gsutil><code>gsutil</code></a></td> <td align=left><code>4.31</code><br></td> <td align=left><a href=https://cloud.google.com/storage/docs/gsutil>Website</a></td> <td align=left>gsutil is a Python application that lets you access Cloud Storage from the command line.</td> </tr> <tr> <td align=left><strong>file transfer</strong></td> <td align=left><a id=lftp><code>lftp</code></a></td> <td align=left><code>4.8.1</code><br></td> <td align=left><a href=https://lftp.yar.ru/ >Website</a></td> <td align=left>LFTP is a sophisticated file transfer program supporting a number of network protocols (ftp, http, sftp, fish, torrent).</td> </tr> <tr> <td align=left><strong>file transfer</strong></td> <td align=left><a id=mpifileutils><code>mpifileutils</code></a></td> <td align=left><code>0.10.1</code><sup><b class=sw_mpi></b></sup><br></td> <td align=left><a href=https://github.com/hpc/mpifileutils>Website</a></td> <td align=left>mpiFileUtils is a suite of <abbr title="Message Passing Interface">MPI</abbr>-based tools to manage large datasets, which may vary from large directory trees to large files.</td> </tr> <tr> <td align=left><strong>file transfer</strong></td> <td align=left><a id=py-globus-cli><code>py-globus-cli</code></a></td> <td align=left><code>1.2.0</code><br><code>1.9.0_py27</code><br><code>1.9.0_py36</code><br></td> <td align=left><a href=https://github.com/globus/globus-cli>Website</a></td> <td align=left>A command line wrapper over the Globus SDK for Python.</td> </tr> <tr> <td align=left><strong>file transfer</strong></td> <td align=left><a id=rclone><code>rclone</code></a></td> <td align=left><code>1.39</code><br><code>1.43.1</code><br><code>1.49.5</code><br><code>1.55.1</code><br></td> <td align=left><a href=https://rclone.org>Website</a></td> <td align=left>Rclone is a command line program to sync files and directories to and from: Google Drive, Amazon S3, Dropbox, Google Cloud Storage, Amazon Drive, Microsoft One Drive, Hubic, Backblaze B2, Yandex Disk, or the local filesystem.</td> </tr> <tr> <td align=left><strong>framework</strong></td> <td align=left><a id=mono><code>mono</code></a></td> <td align=left><code>5.12.0.301</code><br><code>5.20.1.19</code><br></td> <td align=left><a href=https://www.mono-project.com>Website</a></td> <td align=left>Mono is an open source implementation of Microsoft's .NET Framework based on the ECMA standards for C# and the Common Language Runtime.</td> </tr> <tr> <td align=left><strong>language</strong></td> <td align=left><a id=tcltk><code>tcltk</code></a></td> <td align=left><code>8.6.6</code><br></td> <td align=left><a href=https://www.tcl.tk>Website</a></td> <td align=left>Tcl (Tool Command Language) is a dynamic programming language, suitable for web and desktop applications, networking, administration, testing. Tk is a graphical user interface toolkit.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=apr><code>apr</code></a></td> <td align=left><code>1.6.3</code><br></td> <td align=left><a href=https://apr.apache.org/ >Website</a></td> <td align=left>The Apache Portable Runtime is a supporting library for the Apache web server. It provides a set of APIs that map to the underlying operating system.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=apr-util><code>apr-util</code></a></td> <td align=left><code>1.6.1</code><br></td> <td align=left><a href=https://apr.apache.org/ >Website</a></td> <td align=left>The Apache Portable Runtime is a supporting library for the Apache web server. It provides a set of APIs that map to the underlying operating system.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=atk><code>atk</code></a></td> <td align=left><code>2.24.0</code><br></td> <td align=left><a href=https://developer.gnome.org/atk>Website</a></td> <td align=left>ATK is the Accessibility Toolkit. It provides a set of generic interfaces allowing accessibility technologies such as screen readers to interact with a graphical user interface.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=benchmark><code>benchmark</code></a></td> <td align=left><code>1.2.0</code><br></td> <td align=left><a href=https://github.com/google/benchmark>Website</a></td> <td align=left>A microbenchmark support library</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=cairo><code>cairo</code></a></td> <td align=left><code>1.14.10</code><br></td> <td align=left><a href=https://www.cairographics.org/ >Website</a></td> <td align=left>Cairo is a 2D graphics library with support for multiple output devices.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=cups><code>cups</code></a></td> <td align=left><code>2.2.4</code><br></td> <td align=left><a href=https://www.cups.org/ >Website</a></td> <td align=left>CUPS is the standards-based, open source printing system.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=dbus><code>dbus</code></a></td> <td align=left><code>1.10.22</code><br></td> <td align=left><a href=https://www.freedesktop.org/wiki/Software/dbus/ >Website</a></td> <td align=left>D-Bus is a message bus system, a simple way for applications to talk to one another.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=enchant><code>enchant</code></a></td> <td align=left><code>1.6.1</code><br><code>2.2.3</code><br></td> <td align=left><a href=https://abiword.github.io/enchant/ >Website</a></td> <td align=left>Enchant is a library (and command-line program) that wraps a number of different spelling libraries and programs with a consistent interface.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=fltk><code>fltk</code></a></td> <td align=left><code>1.3.4</code><br></td> <td align=left><a href=http://www.fltk.org>Website</a></td> <td align=left>FLTK (pronounced 'fulltick') is a cross-platform C++ GUI toolkit.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=fontconfig><code>fontconfig</code></a></td> <td align=left><code>2.12.4</code><br></td> <td align=left><a href=https://www.freedesktop.org/wiki/Software/fontconfig>Website</a></td> <td align=left>Fontconfig is a library for configuring and customizing font access.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=freeglut><code>freeglut</code></a></td> <td align=left><code>3.0.0</code><br></td> <td align=left><a href=http://freeglut.sourceforge.net/ >Website</a></td> <td align=left>FreeGLUT is a free-software/open-source alternative to the OpenGL Utility Toolkit (GLUT) library.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=freetype><code>freetype</code></a></td> <td align=left><code>2.8</code><br><code>2.9.1</code><br></td> <td align=left><a href=http://freetype.org>Website</a></td> <td align=left>FreeType is a software font engine that is designed to be small, efficient, highly customizable, and portable while capable of producing high-quality output (glyph images).</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=gc><code>gc</code></a></td> <td align=left><code>7.6.0</code><br></td> <td align=left><a href=http://www.hboehm.info/gc>Website</a></td> <td align=left>The Boehm-Demers-Weiser conservative garbage collector can be used as a garbage collecting replacement for C malloc or C++ new.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=gconf><code>gconf</code></a></td> <td align=left><code>2.9.91</code><br></td> <td align=left><a href=https://projects.gnome.org/gconf>Website</a></td> <td align=left>GConf is a system for storing application preferences.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=gdk-pixbuf><code>gdk-pixbuf</code></a></td> <td align=left><code>2.36.8</code><br></td> <td align=left><a href=https://git.gnome.org/browse/gdk-pixbuf>Website</a></td> <td align=left>The GdkPixbuf library provides facilities for loading images in a variety of file formats.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=gflags><code>gflags</code></a></td> <td align=left><code>2.2.1</code><br></td> <td align=left><a href=https://gflags.github.io/gflags/ >Website</a></td> <td align=left>The gflags package contains a C++ library that implements commandline flags processing.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=giflib><code>giflib</code></a></td> <td align=left><code>5.1.4</code><br></td> <td align=left><a href=http://giflib.sourceforge.net/ >Website</a></td> <td align=left>GIFLIB is a package of portable tools and library routines for working with GIF images.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=glib><code>glib</code></a></td> <td align=left><code>2.52.3</code><br></td> <td align=left><a href=https://wiki.gnome.org/Projects/GLib>Website</a></td> <td align=left>The GLib library provides core non-graphical functionality such as high level data types, Unicode manipulation, and an object and type system to C programs.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=glog><code>glog</code></a></td> <td align=left><code>0.3.5</code><br></td> <td align=left><a href=https://github.com/google/glog>Website</a></td> <td align=left>C++ implementation of the Google logging module.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=gnutls><code>gnutls</code></a></td> <td align=left><code>3.5.9</code><br></td> <td align=left><a href=https://www.gnutls.org>Website</a></td> <td align=left>GnuTLS is a secure communications library implementing the SSL, TLS and DTLS protocols and technologies around them.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=gobject-introspection><code>gobject-introspection</code></a></td> <td align=left><code>1.52.1</code><br></td> <td align=left><a href=https://wiki.gnome.org/Projects/GObjectIntrospection>Website</a></td> <td align=left>GObject introspection is a middleware layer between C libraries (using GObject) and language bindings.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=googletest><code>googletest</code></a></td> <td align=left><code>1.8.0</code><br></td> <td align=left><a href=https://github.com/google/googletest>Website</a></td> <td align=left>Google Test is Google's C++ test framework.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=gtk+><code>gtk+</code></a></td> <td align=left><code>2.24.30</code><br><code>3.22.18</code><br></td> <td align=left><a href=https://www.gtk.org>Website</a></td> <td align=left>GTK+, or the GIMP Toolkit, is a multi-platform toolkit for creating graphical user interfaces.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=harfbuzz><code>harfbuzz</code></a></td> <td align=left><code>1.4.8</code><br></td> <td align=left><a href=https://www.freedesktop.org/wiki/Software/HarfBuzz>Website</a></td> <td align=left>HarfBuzz is an OpenType text shaping engine.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=hunspell><code>hunspell</code></a></td> <td align=left><code>1.6.2</code><br></td> <td align=left><a href=https://hunspell.github.io/ >Website</a></td> <td align=left>Hunspell is a spell checker.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=hyphen><code>hyphen</code></a></td> <td align=left><code>2.8.8</code><br></td> <td align=left><a href=https://github.com/hunspell/hyphen>Website</a></td> <td align=left>Hyphen is a hyphenation library to use converted TeX hyphenation patterns.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=icu><code>icu</code></a></td> <td align=left><code>59.1</code><br></td> <td align=left><a href=http://site.icu-project.org>Website</a></td> <td align=left>ICU is a set of C/C++ and Java libraries providing Unicode and Globalization support for software applications.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=json-glib><code>json-glib</code></a></td> <td align=left><code>1.4.4</code><br></td> <td align=left><a href=https://wiki.gnome.org/Projects/JsonGlib>Website</a></td> <td align=left>JSON-GLib is a library providing serialization and deserialization support for the JavaScript Object Notation (JSON) format described by RFC 4627.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libaio><code>libaio</code></a></td> <td align=left><code>0.3.111</code><br></td> <td align=left><a href=https://pagure.io/libaio>Website</a></td> <td align=left>libaio provides the Linux-native API for async <abbr title="Input Output">I/O</abbr>.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libepoxy><code>libepoxy</code></a></td> <td align=left><code>1.4.1</code><br></td> <td align=left><a href=https://github.com/anholt/libepoxy>Website</a></td> <td align=left>Epoxy is a library for handling OpenGL function pointer management for you.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libexif><code>libexif</code></a></td> <td align=left><code>0.6.21</code><br></td> <td align=left><a href=https://github.com/libexif/libexif>Website</a></td> <td align=left>A library for parsing, editing, and saving EXIF data.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libffi><code>libffi</code></a></td> <td align=left><code>3.2.1</code><br></td> <td align=left><a href=http://sourceware.org/libffi>Website</a></td> <td align=left>libffi is a portable Foreign Function Interface library.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libgcrypt><code>libgcrypt</code></a></td> <td align=left><code>1.8.2</code><br></td> <td align=left><a href=https://www.gnupg.org/software/libgcrypt>Website</a></td> <td align=left>Libgcrypt is a general purpose cryptographic library originally based on code from GnuPG.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libgd><code>libgd</code></a></td> <td align=left><code>2.2.5</code><br></td> <td align=left><a href=https://libgd.github.io/ >Website</a></td> <td align=left>GD is an open source code library for the dynamic creation of images by programmers.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libgdiplus><code>libgdiplus</code></a></td> <td align=left><code>5.6</code><br></td> <td align=left><a href=https://github.com/mono/libgdiplus>Website</a></td> <td align=left>C-based implementation of the GDI+ API</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libgpg-error><code>libgpg-error</code></a></td> <td align=left><code>1.27</code><br></td> <td align=left><a href=https://www.gnupg.org/software/libgpg-error>Website</a></td> <td align=left>Libgpg-error is a small library that originally defined common error values for all GnuPG components.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libidl><code>libidl</code></a></td> <td align=left><code>0.8.14</code><br></td> <td align=left><a href=https://github.com/GNOME/libIDL>Website</a></td> <td align=left>The libIDL package contains libraries for Interface Definition Language files. This is a specification for defining portable interfaces.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libjpeg-turbo><code>libjpeg-turbo</code></a></td> <td align=left><code>1.5.1</code><br></td> <td align=left><a href=http://www.libjpeg-turbo.org>Website</a></td> <td align=left>libjpeg-turbo is a JPEG image codec that uses SIMD instructions (MMX, SSE2, AVX2, NEON, AltiVec) to accelerate baseline JPEG compression and decompression on x86, x86-64, ARM, and PowerPC systems</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libmng><code>libmng</code></a></td> <td align=left><code>2.0.3</code><br></td> <td align=left><a href=http://libmng.sourceforge.net>Website</a></td> <td align=left>THE reference library for reading, displaying, writing and examining Multiple-Image Network Graphics. MNG is the animation extension to the popular PNG image-format.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libpng><code>libpng</code></a></td> <td align=left><code>1.2.57</code><br><code>1.6.29</code><br></td> <td align=left><a href=http://libpng.sourceforge.net>Website</a></td> <td align=left>libpng is the official PNG reference library. It supports almost all PNG features, is extensible, and has been extensively tested for over 20 years.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libproxy><code>libproxy</code></a></td> <td align=left><code>0.4.15</code><br></td> <td align=left><a href=https://libproxy.github.io/libproxy/ >Website</a></td> <td align=left>libproxy is a library that provides automatic proxy configuration management.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libressl><code>libressl</code></a></td> <td align=left><code>2.5.3</code><br><code>3.2.1</code><br></td> <td align=left><a href=https://www.libressl.org>Website</a></td> <td align=left>LibreSSL is a version of the TLS/crypto stack forked from OpenSSL in 2014, with goals of modernizing the codebase, improving security, and applying best practice development processes.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libseccomp><code>libseccomp</code></a></td> <td align=left><code>2.3.3</code><br></td> <td align=left><a href=https://github.com/seccomp/libseccomp>Website</a></td> <td align=left>The libseccomp library provides an easy to use, platform independent, interface to the Linux Kernel's syscall filtering mechanism..</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libsodium><code>libsodium</code></a></td> <td align=left><code>1.0.18</code><br></td> <td align=left><a href=https://libsodium.org>Website</a></td> <td align=left>Sodium is a modern, easy-to-use software library for encryption, decryption, signatures, password hashing and more.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libsoup><code>libsoup</code></a></td> <td align=left><code>2.61.2</code><br></td> <td align=left><a href=https://wiki.gnome.org/LibSoup>Website</a></td> <td align=left>libsoup is an HTTP client/server library for GNOME.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libtasn1><code>libtasn1</code></a></td> <td align=left><code>4.13</code><br></td> <td align=left><a href=https://www.gnu.org/software/libtasn1/ >Website</a></td> <td align=left>Libtasn1 is the ASN.1 library used by GnuTLS, p11-kit and some other packages.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libtiff><code>libtiff</code></a></td> <td align=left><code>4.0.8</code><br></td> <td align=left><a href=http://simplesystems.org/libtiff/ >Website</a></td> <td align=left>libtiff provides support for the Tag Image File Format (TIFF), a widely used format for storing image data.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libunistring><code>libunistring</code></a></td> <td align=left><code>0.9.7</code><br></td> <td align=left><a href=https://www.gnu.org/software/libunistring/ >Website</a></td> <td align=left>Libunistring provides functions for manipulating Unicode strings and for manipulating C strings according to the Unicode standard.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libuuid><code>libuuid</code></a></td> <td align=left><code>1.0.3</code><br></td> <td align=left><a href=https://sourceforge.net/projects/libuuid>Website</a></td> <td align=left>Portable uuid C library.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libuv><code>libuv</code></a></td> <td align=left><code>1.38.1</code><br></td> <td align=left><a href=https://github.com/libuv/libuv>Website</a></td> <td align=left>libuv is a multi-platform support library with a focus on asynchronous <abbr title="Input Output">I/O</abbr>.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libwebp><code>libwebp</code></a></td> <td align=left><code>0.6.1</code><br></td> <td align=left><a href=https://developers.google.com/speed/webp>Website</a></td> <td align=left>WebP is a modern image format that provides superior lossless and lossy compression for images on the web.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libxkbcommon><code>libxkbcommon</code></a></td> <td align=left><code>0.9.1</code><br></td> <td align=left><a href=https://github.com/xkbcommon/libxkbcommon>Website</a></td> <td align=left>libxkbcommon is a keyboard keymap compiler and support library which processes a reduced subset of keymaps as defined by the XKB (X Keyboard Extension) specification.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libxml2><code>libxml2</code></a></td> <td align=left><code>2.9.4</code><br></td> <td align=left><a href=http://xmlsoft.org/ >Website</a></td> <td align=left>Libxml2 is a XML C parser and toolkit.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=libxslt><code>libxslt</code></a></td> <td align=left><code>1.1.32</code><br></td> <td align=left><a href=http://xmlsoft.org/libxslt>Website</a></td> <td align=left>Libxslt is the XSLT C library developed for the GNOME project. XSLT itself is a an XML language to define transformation for XML.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=mesa><code>mesa</code></a></td> <td align=left><code>17.1.6</code><br></td> <td align=left><a href=https://www.mesa3d.org>Website</a></td> <td align=left>Mesa is an open-source implementation of the OpenGL, Vulkan and other specifications.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=ncurses><code>ncurses</code></a></td> <td align=left><code>6.0</code><br></td> <td align=left><a href=https://www.gnu.org/software/ncurses>Website</a></td> <td align=left>The ncurses (new curses) library is a free software emulation of curses in System V Release 4.0 (SVr4), and more.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=nettle><code>nettle</code></a></td> <td align=left><code>3.3</code><br></td> <td align=left><a href=http://www.lysator.liu.se/~nisse/nettle/ >Website</a></td> <td align=left>Nettle is a cryptographic library that is designed to fit easily in more or less any context.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=openjpeg><code>openjpeg</code></a></td> <td align=left><code>2.3.1</code><br></td> <td align=left><a href=https://github.com/uclouvain/openjpeg>Website</a></td> <td align=left>OpenJPEG is an open-source JPEG 2000 codec written in C language.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=orbit><code>orbit</code></a></td> <td align=left><code>2.14.19</code><br></td> <td align=left><a href=https://projects.gnome.org/ORBit2>Website</a></td> <td align=left>ORBit2 is a CORBA 2.4-compliant Object Request Broker (ORB) featuring mature C, C++ and Python bindings.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=pango><code>pango</code></a></td> <td align=left><code>1.40.10</code><br></td> <td align=left><a href=http://www.pango.org>Website</a></td> <td align=left>Pango is a library for laying out and rendering of text, with an emphasis on internationalization.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=pcre><code>pcre</code></a></td> <td align=left><code>8.40</code><br></td> <td align=left><a href=http://www.pcre.org/ >Website</a></td> <td align=left>The PCRE library is a set of functions that implement regular expression pattern matching using the same syntax and semantics as Perl 5.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=pcre2><code>pcre2</code></a></td> <td align=left><code>10.35</code><br></td> <td align=left><a href=http://www.pcre.org/ >Website</a></td> <td align=left>The PCRE22 library is a set of functions that implement regular expression pattern matching using the same syntax and semantics as Perl 5.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=popt><code>popt</code></a></td> <td align=left><code>1.16</code><br></td> <td align=left><a href=http://rpm5.org/files/popt>Website</a></td> <td align=left>Library for parsing command line options.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=py-lmdb><code>py-lmdb</code></a></td> <td align=left><code>0.93</code><br></td> <td align=left><a href=https://lmdb.readthedocs.io/en/release/ >Website</a></td> <td align=left>Universal Python binding for the LMDB 'Lightning' Database.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=py-mako><code>py-mako</code></a></td> <td align=left><code>1.0.7_py27</code><sup><b class=sw_def></b></sup><br><code>1.0.7_py36</code><br></td> <td align=left><a href=http://www.makotemplates.org/ >Website</a></td> <td align=left>Mako is a template library written in Python. It provides a familiar, non-XML syntax which compiles into Python modules for maximum performance.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=py-pygobject><code>py-pygobject</code></a></td> <td align=left><code>3.32.2_py36</code><br></td> <td align=left><a href=https://pygobject.readthedocs.io/ >Website</a></td> <td align=left>PyGObject is a Python package which provides bindings for GObject based libraries such as GTK, GStreamer, WebKitGTK, GLib, GIO and many more.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=py-pyopengl><code>py-pyopengl</code></a></td> <td align=left><code>3.1.5_py39</code><br></td> <td align=left><a href=http://pyopengl.sourceforge.net/ >Website</a></td> <td align=left>Standard OpenGL bindings for Python.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=py-pyqt5><code>py-pyqt5</code></a></td> <td align=left><code>5.9.1_py36</code><br></td> <td align=left><a href=https://www.riverbankcomputing.com/software/pyqt/ >Website</a></td> <td align=left>PyQt5 is a comprehensive set of Python bindings for Qt v5.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=readline><code>readline</code></a></td> <td align=left><code>7.0</code><br></td> <td align=left><a href=https://cnswww.cns.cwru.edu/php/chet/readline/rltop.html>Website</a></td> <td align=left>The GNU Readline library provides a set of functions for use by applications that allow users to edit command lines as they are typed in.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=serf><code>serf</code></a></td> <td align=left><code>1.3.9</code><br></td> <td align=left><a href=https://serf.apache.org/ >Website</a></td> <td align=left>The serf library is a high performance C-based HTTP client library built upon the Apache Portable Runtime (APR) library.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=snappy><code>snappy</code></a></td> <td align=left><code>1.1.7</code><br></td> <td align=left><a href=https://github.com/google/snappy>Website</a></td> <td align=left>A fast compressor/decompressor.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=talloc><code>talloc</code></a></td> <td align=left><code>2.1.14</code><br></td> <td align=left><a href=https://talloc.samba.org>Website</a></td> <td align=left>talloc is a hierarchical, reference counted memory pool system with destructors.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=utf8proc><code>utf8proc</code></a></td> <td align=left><code>2.4.0</code><br></td> <td align=left><a href=https://juliastrings.github.io/utf8proc/ >Website</a></td> <td align=left>iutf8proc is a small, clean C library that provides Unicode normalization, case-folding, and other operations for data in the UTF-8 encoding.</td> </tr> <tr> <td align=left><strong>libs</strong></td> <td align=left><a id=wxwidgets><code>wxwidgets</code></a></td> <td align=left><code>3.0.4</code><br></td> <td align=left><a href=https://www.wxwidgets.org/ >Website</a></td> <td align=left>wxWidgets is a C++ library that lets developers create applications for Windows, macOS, Linux and other platforms with a single code base.</td> </tr> <tr> <td align=left><strong>media</strong></td> <td align=left><a id=ffmpeg><code>ffmpeg</code></a></td> <td align=left><code>4.0</code><br><code>4.2.1</code><br></td> <td align=left><a href=https://www.ffmpeg.org>Website</a></td> <td align=left>FFmpeg is the leading multimedia framework, able to decode, encode, transcode, mux, demux, stream, filter and play pretty much anything that humans and machines have created.</td> </tr> <tr> <td align=left><strong>media</strong></td> <td align=left><a id=libsndfile><code>libsndfile</code></a></td> <td align=left><code>1.0.28</code><br></td> <td align=left><a href=http://www.mega-nerd.com/libsndfile>Website</a></td> <td align=left>Libsndfile is a C library for reading and writing files containing sampled sound (such as MS Windows WAV and the Apple/SGI AIFF format) through one standard library interface.</td> </tr> <tr> <td align=left><strong>performance</strong></td> <td align=left><a id=likwid><code>likwid</code></a></td> <td align=left><code>4.3.2</code><br></td> <td align=left><a href=https://github.com/RRZE-HPC/likwid>Website</a></td> <td align=left>Likwid is a simple toolsuite of command line applications for performance oriented programmers.</td> </tr> <tr> <td align=left><strong>resource monitoring</strong></td> <td align=left><a id=nvtop><code>nvtop</code></a></td> <td align=left><code>1.1.0</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://github.com/Syllo/nvtop>Website</a></td> <td align=left>Nvtop stands for NVidia TOP, a (h)top like task monitor for NVIDIA GPUs.</td> </tr> <tr> <td align=left><strong>scm</strong></td> <td align=left><a id=git><code>git</code></a></td> <td align=left><code>2.12.2</code><br></td> <td align=left><a href=http://git-scm.com>Website</a></td> <td align=left>Git is a free and open source distributed version control system designed to handle everything from small to very large projects with speed and efficiency.</td> </tr> <tr> <td align=left><strong>scm</strong></td> <td align=left><a id=git-lfs><code>git-lfs</code></a></td> <td align=left><code>2.4.0</code><br></td> <td align=left><a href=https://git-lfs.github.com/ >Website</a></td> <td align=left>Git Large File Storage (LFS) replaces large files such as audio samples, videos, datasets, and graphics with text pointers inside Git, while storing the file contents on a remote server.</td> </tr> <tr> <td align=left><strong>scm</strong></td> <td align=left><a id=libgit2><code>libgit2</code></a></td> <td align=left><code>1.1.0</code><br></td> <td align=left><a href=https://libgit2.org/ >Website</a></td> <td align=left>libgit2 is a portable, pure C implementation of the Git core methods provided as a re-entrant linkable library with a solid API</td> </tr> <tr> <td align=left><strong>scm</strong></td> <td align=left><a id=mercurial><code>mercurial</code></a></td> <td align=left><code>4.5.3</code><br></td> <td align=left><a href=https://www.mercurial-scm.org/ >Website</a></td> <td align=left>Mercurial is a free, distributed source control management tool.</td> </tr> <tr> <td align=left><strong>scm</strong></td> <td align=left><a id=py-dvc><code>py-dvc</code></a></td> <td align=left><code>0.91.1_py36</code><br></td> <td align=left><a href=https://dvc.org/ >Website</a></td> <td align=left>Data Version Control or DVC is an open-source tool for data science and machine learning projects.</td> </tr> <tr> <td align=left><strong>scm</strong></td> <td align=left><a id=subversion><code>subversion</code></a></td> <td align=left><code>1.9.7</code><br><code>1.12.2</code><br></td> <td align=left><a href=https://subversion.apache.org/ >Website</a></td> <td align=left>Subversion is an open source version control system.</td> </tr> <tr> <td align=left><strong>tools</strong></td> <td align=left><a id=clinfo><code>clinfo</code></a></td> <td align=left><code>2.2.18.04.06</code><sup><b class=sw_gpu></b></sup><br></td> <td align=left><a href=https://github.com/Oblomov/clinfo>Website</a></td> <td align=left>clinfo is a simple command-line application that enumerates all possible (known) properties of the OpenCL platform and devices available on the system.</td> </tr> <tr> <td align=left><strong>tools</strong></td> <td align=left><a id=curl><code>curl</code></a></td> <td align=left><code>7.54.0</code><br></td> <td align=left><a href=https://curl.haxx.se/ >Website</a></td> <td align=left>curl is an open source command line tool and library for transferring data with URL syntax.</td> </tr> <tr> <td align=left><strong>tools</strong></td> <td align=left><a id=depot_tools><code>depot_tools</code></a></td> <td align=left><code>20200731</code><br></td> <td align=left><a href=https://chromium.googlesource.com/chromium/tools/depot_tools.git>Website</a></td> <td align=left>Tools for working with Chromium development.</td> </tr> <tr> <td align=left><strong>tools</strong></td> <td align=left><a id=expat><code>expat</code></a></td> <td align=left><code>2.2.3</code><br></td> <td align=left><a href=http://libexpat.github.io>Website</a></td> <td align=left>Expat is a stream-oriented XML parser library written in C.</td> </tr> <tr> <td align=left><strong>tools</strong></td> <td align=left><a id=graphicsmagick><code>graphicsmagick</code></a></td> <td align=left><code>1.3.26</code><br></td> <td align=left><a href=http://www.graphicsmagick.org/ >Website</a></td> <td align=left>GraphicsMagick is the swiss army knife of image processing.</td> </tr> <tr> <td align=left><strong>tools</strong></td> <td align=left><a id=imagemagick><code>imagemagick</code></a></td> <td align=left><code>7.0.7-2</code><br></td> <td align=left><a href=https://www.imagemagick.org>Website</a></td> <td align=left>ImageMagick is a free and open-source software suite for displaying, converting, and editing raster image and vector image files.</td> </tr> <tr> <td align=left><strong>tools</strong></td> <td align=left><a id=leveldb><code>leveldb</code></a></td> <td align=left><code>1.20</code><br></td> <td align=left><a href=https://symas.com/lightning-memory-mapped-database/ >Website</a></td> <td align=left>Symas LMDB is an extraordinarily fast, memory-efficient database we developed for the Symas OpenLDAP Project.</td> </tr> <tr> <td align=left><strong>tools</strong></td> <td align=left><a id=lmdb><code>lmdb</code></a></td> <td align=left><code>0.9.21</code><br></td> <td align=left><a href=https://symas.com/lightning-memory-mapped-database/ >Website</a></td> <td align=left>Symas LMDB is an extraordinarily fast, memory-efficient database we developed for the Symas OpenLDAP Project.</td> </tr> <tr> <td align=left><strong>tools</strong></td> <td align=left><a id=motif><code>motif</code></a></td> <td align=left><code>2.3.7</code><br></td> <td align=left><a href=https://motif.ics.com>Website</a></td> <td align=left>Motif is the toolkit for the Common Desktop Environment.</td> </tr> <tr> <td align=left><strong>tools</strong></td> <td align=left><a id=parallel><code>parallel</code></a></td> <td align=left><code>20180122</code><br><code>20200822</code><br></td> <td align=left><a href=https://www.gnu.org/software/parallel/ >Website</a></td> <td align=left>GNU parallel is a shell tool for executing jobs in parallel using one or more computers.</td> </tr> <tr> <td align=left><strong>tools</strong></td> <td align=left><a id=qt><code>qt</code></a></td> <td align=left><code>5.9.1</code><br></td> <td align=left><a href=https://www.qt.io>Website</a></td> <td align=left>QT is a cross-platform application framework that is used for developing application software that can be run on various software and hardware platforms.</td> </tr> <tr> <td align=left><strong>tools</strong></td> <td align=left><a id=ripgrep><code>ripgrep</code></a></td> <td align=left><code>11.0.1</code><br></td> <td align=left><a href=https://github.com/BurntSushi/ripgrep>Website</a></td> <td align=left>ripgrep recursively searches directories for a regex pattern.</td> </tr> <tr> <td align=left><strong>tools</strong></td> <td align=left><a id=rocksdb><code>rocksdb</code></a></td> <td align=left><code>5.7.3</code><br></td> <td align=left><a href=https://rocksdb.org>Website</a></td> <td align=left>A library that provides an embeddable, persistent key-value store for fast storage.</td> </tr> <tr> <td align=left><strong>tools</strong></td> <td align=left><a id=x11><code>x11</code></a></td> <td align=left><code>7.7</code><br></td> <td align=left><a href=https://www.x.org>Website</a></td> <td align=left>The X.Org project provides an open source implementation of the X Window System.</td> </tr> <tr> <td align=left><strong>tools</strong></td> <td align=left><a id=xkeyboard-config><code>xkeyboard-config</code></a></td> <td align=left><code>2.21</code><br></td> <td align=left><a href=https://www.freedesktop.org/wiki/Software/XKeyboardConfig/ >Website</a></td> <td align=left>The non-arch keyboard configuration database for X Window.</td> </tr> </tbody> </table> <h3 id=viz><strong>viz</strong><a class=headerlink href=#viz title="Permanent link">#</a></h3> <table> <thead> <tr> <th align=left>Field</th> <th align=left><img style=float:left;min-width:110px;visibility:hidden>Module&nbsp;name</th> <th align=left><img style=float:left;min-width:90px;visibility:hidden>Version(s)</th> <th align=left>URL</th> <th align=left>Description</th> </tr> </thead> <tbody> <tr> <td align=left><strong>data</strong></td> <td align=left><a id=ncview><code>ncview</code></a></td> <td align=left><code>2.1.7</code><br></td> <td align=left><a href=http://meteora.ucsd.edu/~pierce/ncview_home_page.html>Website</a></td> <td align=left>Ncview is a visual browser for netCDF format files.</td> </tr> <tr> <td align=left><strong>gis</strong></td> <td align=left><a id=panoply><code>panoply</code></a></td> <td align=left><code>4.10.8</code><br></td> <td align=left><a href=https://www.giss.nasa.gov/tools/panoply/ >Website</a></td> <td align=left>Panoply plots geo-referenced and other arrays from netCDF, HDF, GRIB, and other datasets.</td> </tr> <tr> <td align=left><strong>graphs</strong></td> <td align=left><a id=graphviz><code>graphviz</code></a></td> <td align=left><code>2.40.1</code><br><code>2.44.1</code><br></td> <td align=left><a href=https://www.graphviz.org/ >Website</a></td> <td align=left>Graphviz is open source graph visualization software.</td> </tr> <tr> <td align=left><strong>imaging</strong></td> <td align=left><a id=py-pillow><code>py-pillow</code></a></td> <td align=left><code>5.1.0_py27</code><sup><b class=sw_def></b></sup><br><code>5.1.0_py36</code><br><code>7.0.0_py36</code><br><code>8.2.0_py39</code><br></td> <td align=left><a href=https://python-pillow.org>Website</a></td> <td align=left>Pillow is a friendly PIL (Python Imaging Library) fork.</td> </tr> <tr> <td align=left><strong>imaging</strong></td> <td align=left><a id=py-pillow-simd><code>py-pillow-simd</code></a></td> <td align=left><code>7.0.0.post3_py36</code><br></td> <td align=left><a href=https://github.com/uploadcare/pillow-simd>Website</a></td> <td align=left>Pillow-SIMD is an optimized version of Pillow</td> </tr> <tr> <td align=left><strong>molecular visualization</strong></td> <td align=left><a id=pymol><code>pymol</code></a></td> <td align=left><code>1.8.6.2</code><sup><b class=sw_lic></b></sup><br></td> <td align=left><a href=https://pymol.org>Website</a></td> <td align=left>PyMOL is a Python-enhanced molecular graphics tool.</td> </tr> <tr> <td align=left><strong>plotting</strong></td> <td align=left><a id=gnuplot><code>gnuplot</code></a></td> <td align=left><code>5.2.0</code><br></td> <td align=left><a href=http://www.gnuplot.info>Website</a></td> <td align=left>Gnuplot is a portable command-line driven graphing utility for Linux, <abbr title="Operating System">OS</abbr>/2, MS Windows, OSX, VMS, and many other platforms.</td> </tr> <tr> <td align=left><strong>plotting</strong></td> <td align=left><a id=grace><code>grace</code></a></td> <td align=left><code>5.1.25</code><br></td> <td align=left><a href=http://plasma-gate.weizmann.ac.il/Grace>Website</a></td> <td align=left>Grace is a WYSIWYG tool to make two-dimensional plots of numerical data.</td> </tr> <tr> <td align=left><strong>plotting</strong></td> <td align=left><a id=py-basemap><code>py-basemap</code></a></td> <td align=left><code>1.1.0_py27</code><sup><b class=sw_def></b></sup><br><code>1.1.0_py36</code><br></td> <td align=left><a href=https://matplotlib.org/basemap/ >Website</a></td> <td align=left>The matplotlib basemap toolkit is a library for plotting 2D data on maps in Python.</td> </tr> <tr> <td align=left><strong>plotting</strong></td> <td align=left><a id=py-matplotlib><code>py-matplotlib</code></a></td> <td align=left><code>2.2.2_py27</code><sup><b class=sw_def></b></sup><br><code>2.1.2_py27</code><br><code>2.1.2_py36</code><br><code>2.2.2_py36</code><br><code>3.1.1_py36</code><br><code>3.2.1_py36</code><br><code>3.4.2_py39</code><br></td> <td align=left><a href=http://matplotlib.org>Website</a></td> <td align=left>Matplotlib is a Python 2D plotting library which produces publication quality figures in a variety of hardcopy formats and interactive environments across platforms.</td> </tr> <tr> <td align=left><strong>plotting</strong></td> <td align=left><a id=py-plotly><code>py-plotly</code></a></td> <td align=left><code>2.4.1_py27</code><br></td> <td align=left><a href=https://plot.ly/python/ >Website</a></td> <td align=left>Plotly's Python graphing library makes interactive, publication-quality graphs online.</td> </tr> <tr> <td align=left><strong>remote display</strong></td> <td align=left><a id=virtualgl><code>virtualgl</code></a></td> <td align=left><code>2.5.2</code><br></td> <td align=left><a href=http://www.virtualgl.org>Website</a></td> <td align=left>VirtualGL is an open source toolkit that gives any Unix or Linux remote display software the ability to run OpenGL applications with full 3D hardware acceleration.</td> </tr> </tbody> </table> </article> </div> </div> </main> <!-- Application footer --> <footer class=md-footer> <!-- Link to previous and/or next page --> <div class=md-footer-nav> <nav class="md-footer-nav__inner md-grid" aria-label=Footer> <!-- Link to previous page --> <a href=../overview/ title=Overview class="md-footer-nav__link md-footer-nav__link--prev" rel=prev> <div class="md-footer-nav__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg> </div> <div class=md-footer-nav__title> <div class=md-ellipsis> <span class=md-footer-nav__direction> Previous </span> Overview </div> </div> </a> <!-- Link to next page --> <a href=../modules/ title=Modules class="md-footer-nav__link md-footer-nav__link--next" rel=next> <div class=md-footer-nav__title> <div class=md-ellipsis> <span class=md-footer-nav__direction> Next </span> Modules </div> </div> <div class="md-footer-nav__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg> </div> </a> </nav> </div> <!-- Further information --> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <!-- Copyright and theme information --> <div class=md-footer-copyright> <div class=md-footer-copyright__highlight> Copyright &copy; 2014 - 2021 &mdash; <a href=mailto:srcc-support@stanford.edu>Stanford Research Computing Center</a> </div> Made with <span class="twemoji pulsate"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.27 2 8.5 2 5.41 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.08C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.41 22 8.5c0 3.77-3.4 6.86-8.55 11.53L12 21.35z"/></svg> </span> in <span class=md-footer-copyright__highlight> <a href=https://www.google.com/maps/place/Stanford,+CA>Stanford, California</a> </span> </div> <!-- Social links --> <div class=md-footer-social> <a href=https://srcc.stanford.edu target=_blank rel=noopener title="SRCC website" class=md-footer-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><path d="M248 8C111.03 8 0 119.03 0 256s111.03 248 248 248 248-111.03 248-248S384.97 8 248 8zm82.29 357.6c-3.9 3.88-7.99 7.95-11.31 11.28-2.99 3-5.1 6.7-6.17 10.71-1.51 5.66-2.73 11.38-4.77 16.87l-17.39 46.85c-13.76 3-28 4.69-42.65 4.69v-27.38c1.69-12.62-7.64-36.26-22.63-51.25-6-6-9.37-14.14-9.37-22.63v-32.01c0-11.64-6.27-22.34-16.46-27.97-14.37-7.95-34.81-19.06-48.81-26.11-11.48-5.78-22.1-13.14-31.65-21.75l-.8-.72a114.792 114.792 0 01-18.06-20.74c-9.38-13.77-24.66-36.42-34.59-51.14 20.47-45.5 57.36-82.04 103.2-101.89l24.01 12.01C203.48 89.74 216 82.01 216 70.11v-11.3c7.99-1.29 16.12-2.11 24.39-2.42l28.3 28.3c6.25 6.25 6.25 16.38 0 22.63L264 112l-10.34 10.34c-3.12 3.12-3.12 8.19 0 11.31l4.69 4.69c3.12 3.12 3.12 8.19 0 11.31l-8 8a8.008 8.008 0 01-5.66 2.34h-8.99c-2.08 0-4.08.81-5.58 2.27l-9.92 9.65a8.008 8.008 0 00-1.58 9.31l15.59 31.19c2.66 5.32-1.21 11.58-7.15 11.58h-5.64c-1.93 0-3.79-.7-5.24-1.96l-9.28-8.06a16.017 16.017 0 00-15.55-3.1l-31.17 10.39a11.95 11.95 0 00-8.17 11.34c0 4.53 2.56 8.66 6.61 10.69l11.08 5.54c9.41 4.71 19.79 7.16 30.31 7.16s22.59 27.29 32 32h66.75c8.49 0 16.62 3.37 22.63 9.37l13.69 13.69a30.503 30.503 0 018.93 21.57 46.536 46.536 0 01-13.72 32.98zM417 274.25c-5.79-1.45-10.84-5-14.15-9.97l-17.98-26.97a23.97 23.97 0 010-26.62l19.59-29.38c2.32-3.47 5.5-6.29 9.24-8.15l12.98-6.49C440.2 193.59 448 223.87 448 256c0 8.67-.74 17.16-1.82 25.54L417 274.25z"/></svg> </a> <a href=https://github.com/stanford-rc target=_blank rel=noopener title="SRCC on GitHub" class=md-footer-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg> </a> <a href=https://twitter.com/StanfordCompute target=_blank rel=noopener title="SRCC on Twitter" class=md-footer-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 512 512"><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg> </a> <a href=https://srcc.slack.com/ target=_blank rel=noopener title="SRCC Users Slack workspace" class=md-footer-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><path d="M94.12 315.1c0 25.9-21.16 47.06-47.06 47.06S0 341 0 315.1c0-25.9 21.16-47.06 47.06-47.06h47.06v47.06zm23.72 0c0-25.9 21.16-47.06 47.06-47.06s47.06 21.16 47.06 47.06v117.84c0 25.9-21.16 47.06-47.06 47.06s-47.06-21.16-47.06-47.06V315.1zm47.06-188.98c-25.9 0-47.06-21.16-47.06-47.06S139 32 164.9 32s47.06 21.16 47.06 47.06v47.06H164.9zm0 23.72c25.9 0 47.06 21.16 47.06 47.06s-21.16 47.06-47.06 47.06H47.06C21.16 243.96 0 222.8 0 196.9s21.16-47.06 47.06-47.06H164.9zm188.98 47.06c0-25.9 21.16-47.06 47.06-47.06 25.9 0 47.06 21.16 47.06 47.06s-21.16 47.06-47.06 47.06h-47.06V196.9zm-23.72 0c0 25.9-21.16 47.06-47.06 47.06-25.9 0-47.06-21.16-47.06-47.06V79.06c0-25.9 21.16-47.06 47.06-47.06 25.9 0 47.06 21.16 47.06 47.06V196.9zM283.1 385.88c25.9 0 47.06 21.16 47.06 47.06 0 25.9-21.16 47.06-47.06 47.06-25.9 0-47.06-21.16-47.06-47.06v-47.06h47.06zm0-23.72c-25.9 0-47.06-21.16-47.06-47.06 0-25.9 21.16-47.06 47.06-47.06h117.84c25.9 0 47.06 21.16 47.06 47.06 0 25.9-21.16 47.06-47.06 47.06H283.1z"/></svg> </a> </div> </div> </div> </footer> </div> <script src=../../../assets/javascripts/vendor.18f0862e.min.js></script> <script src=../../../assets/javascripts/bundle.994580cf.min.js></script><script id=__lang type=application/json>{"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing"}</script> <script>
        app = initialize({
          base: "../../..",
          features: ['navigation.instant'],
          search: Object.assign({
            worker: "../../../assets/javascripts/worker/search.9c0e82ba.min.js"
          }, typeof search !== "undefined" && search)
        })
      </script> </body> </html>